---
title: "Analysis of bulk RNA-seq data"
author:
  - name: Axel Klenk
    affiliation: Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona
    email: axelvolker.klenk@upf.edu
  - name: Robert Castelo
    affiliation: Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona
    email: robert.castelo@upf.edu
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Analysis of bulk RNA-seq data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: GSVAEuroBioC2024.bib
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Importing bulk RNA-seq data into a `SummarizedExperiment` object

Here we will use the bulk RNA-seq data from [@costa2021genome].

```{r}
fname <- "CostaEtAl2021_FIRDBS_bulkRNAseq_data_ENTREZID.csv.gz"
expr <- read.csv(system.file(file.path("extdata", fname), package="GSVAEuroBioC2024"),
                 row.names=1)
dim(expr)
expr[1:5, 1:5]
fname <- "CostaEtAl2021_FIRDBS_phenotype_data.csv.gz"
pdat <- read.csv(system.file(file.path("extdata", fname), package="GSVAEuroBioC2024"),
                 row.names=1)
dim(pdat)
head(pdat)
table(pdat$FIR, pdat$Sex)
```

Build a `SummarizedExperiment` object.

```{r, message=FALSE}
library(GSEABase)
library(SummarizedExperiment)

stopifnot(all(colnames(expr) == rownames(pdat))) ## QC
se <- SummarizedExperiment(assay=list(logCPM=as.matrix(expr)),
                           colData=pdat)
gsvaAnnotation(se) <- ENSEMBLIdentifier("org.Hs.eg.db")
se
```

# Importing gene sets

```{r, message=FALSE}
library(GSVA)

URL <- "https://data.broadinstitute.org/gsea-msigdb/msigdb/release/2023.2.Hs/c7.immunesigdb.v2023.2.Hs.symbols.gmt"
genesets <- readGMT(URL, geneIdType=SymbolIdentifier())
genesets
```

# How to run GSVA

# Session information

```{r}
sessionInfo()
```

# References
