---
title: "Analysis of single-cell RNA-seq data"
author:
  - name: Axel Klenk
    affiliation: Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona
    email: axelvolker.klenk@upf.edu
  - name: Robert Castelo
    affiliation: Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona
    email: robert.castelo@upf.edu
date: "`r Sys.Date()`" 
output:
  bookdown::html_document2:
    base_format: rmarkdown::html_vignette
    fig_caption: yes
    toc: true
pkgdown:
  as_is: true
bibliography: GSVAEuroBioC2024.bib
link-citations: yes
vignette: >
  %\VignetteIndexEntry{Analysis of single-cell RNA-seq data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.align="center",
  fig.wide=TRUE
)

## from http://yihui.name/knitr/hooks#chunk_hooks
knitr::knit_hooks$set(small.mar=function(before, options, envir) {
                                  ## smaller margin on top and right
                                  if (before)
                                    par(mar=c(4, 5, 1, 1))
                                })
```

Vignette built using `GSVA`, version `r packageDescription("GSVA")[["Version"]]`.

# Importing scRNA-seq data into a `SingleCellExperiment` object

```{r, message=FALSE}
library(TENxPBMCData)

sce <- TENxPBMCData(dataset="pbmc4k")
sce
```

# Importing gene sets

```{r}
library(GSVA)

fname <- "PBMC_gene_sets_TrianaEtAl2021.gmt"
gsets <- readGMT(system.file("extdata", fname, package="GSVAEuroBioC2024"))

```
# How to run GSVA

# Session information

```{r}
sessionInfo()
```

# References
