---
title: "Analysis of single-cell RNA-seq data"
author:
  - name: Axel Klenk
    affiliation: Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona
    email: axelvolker.klenk@upf.edu
  - name: Robert Castelo
    affiliation: Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona
    email: robert.castelo@upf.edu
date: "`r Sys.Date()`" 
output:
  bookdown::html_document2:
    base_format: rmarkdown::html_vignette
    fig_caption: yes
    toc: true
pkgdown:
  as_is: true
bibliography: GSVAEuroBioC2024.bib
link-citations: yes
vignette: >
  %\VignetteIndexEntry{Analysis of single-cell RNA-seq data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.align="center",
  fig.wide=TRUE
)

## from http://yihui.name/knitr/hooks#chunk_hooks
knitr::knit_hooks$set(small.mar=function(before, options, envir) {
                                  ## smaller margin on top and right
                                  if (before)
                                    par(mar=c(4, 5, 1, 1))
                                })
```

Vignette built using `GSVA`, version `r packageDescription("GSVA")[["Version"]]`.

# Importing scRNA-seq data into a `SingleCellExperiment` object

In this vignette we will use the
[TENxPBMCData](https://bioconductor.org/packages/TENxPBMCData) to download a
[10X Genomics](https://support.10xgenomics.com/single-cell-gene-expression/datasets)
single-cell RNA-seq data set generated from peripheral blood mononuclear cells
(PBMC), into a
[SingleCellExperiment](https://bioconductor.org/packages/SingleCellExperiment)
object.

```{r, message=FALSE, warning=FALSE}
library(SingleCellExperiment)
library(TENxPBMCData)

sce <- TENxPBMCData(dataset="pbmc4k")
sce
```
This single-cell dataset consists of raw counts for `r nrow(sce)` genes by
`r ncol(sce)` cells.

```{r}
assay(sce)[1:5, 1:8]
```

# Pre-processing of scRNA-seq data

Here we do the pre-processing of the scRNA-seq dataset using the
Bioconductor package [scuttle](https://bioconductor.org/packages/scuttle). We
start by identifying mitochondrial genes.

```{r, message=FALSE, warning=FALSE}
library(scuttle)

is_mito <- grepl("^MT-", rowData(sce)$Symbol_TENx)
table(is_mito) 
```
Now calculate quality control (QC) metrics and filter out low-quality cells.

```{r}
sce <- quickPerCellQC(sce, subsets=list(Mito=is_mito),
                      sub.fields="subsets_Mito_percent")
dim(sce)
```
Figure \@ref(fig:cntxgene) below shows the empirical cumulative distribution of
counts per gene in logarithmic scale.

```{r cntxgene, echo=TRUE, small.mar=TRUE, fig.height=4, fig.width=4, units="in", dpi=100, fig.cap="Filtering lowly-expressed genes. Empirical cumulative distribution of UMI counts per gene. The red vertical bar indicates a cutoff value of 100 UMI counts per gene across all cells, below which genes will be filtered out."}
cntxgene <- rowSums(assays(sce)$counts)+1
plot.ecdf(cntxgene, xaxt="n", panel.first=grid(), xlab="UMI counts per gene",
                    log="x", main="", xlim=c(1, 1e5), las=1)
axis(1, at=10^(0:5), labels=10^(0:5))
abline(v=100, lwd=2, col="red")
```
We filter out lowly-expressed genes, by selecting those with at least 100 UMI
counts across all cells.

```{r}
sce <- sce[cntxgene >= 100, ]
dim(sce)
```
Finally, we calculate library size factors and normalized log units of
expression.

```{r}
sce <- computeLibraryFactors(sce)
sce <- logNormCounts(sce)
assayNames(sce)
```
The assay `logcounts` contains the filtered and normalized log units of
expression that we should use as input to GSVA.

# Importing gene sets

Here we read a GMT file storing a collection of 22 gene sets used by
@diaz2019evaluation to benchmark methods for cell type assignment in PBMC
scRNA-seq data.

```{r, message=FALSE, warning=FALSE}
library(GSEABase)
library(GSVA)

fname <- system.file("extdata", "pbmc_cell_type_gene_set_signatures.gmt.gz",
                     package="GSVAEuroBioC2024")
genesets <- readGMT(fname, geneIdType=SymbolIdentifier())
genesets
```

# Running GSVA on a `SingleCellExperiment` object

GSVA can take as input a `SingleCellExperiment` object, but before we run
GSVA we should take a couple of steps and considerations. In this vignette,
genes in that object are defined by
[Ensembl stable gene identifiers](https://www.ensembl.org/info/genome/stable_ids/index.html),
i.e., they start with `ENSG`, while gene identifiers in the collection of genes
we imported in the previous section correspond to
[HUGO gene symbols](https://www.genenames.org). To facilitate that GSVA
internally maps the gene symbols in gene sets to the corresponding Ensembl gene
identifiers of the scRNA-seq data, we should add the following metadata to the
`SingleCellExperiment` object.

```{r}
gsvaAnnotation(sce) <- ENSEMBLIdentifier("org.Hs.eg.db")
```
The assay data is stored as a sparse `DelayedMatrix` object using an `HDF5`
backend. However, GSVA currently[^footnote-1] only provides support for sparse
matrices stored in `dgCMatrix` objects.

[^footnote-1]: We expect to provide support in the near future for sparse
  on-disk matrices, such as sparse `DelayedMatrix` objects.

This means that we need to coerce the assay we intend to use as input for
GSVA to a `dgCMatrix`, as follows.

```{r}
assays(sce)$logcounts <- as(assays(sce)$logcounts, "dgCMatrix")
```
Independently of this previous issue, one should be aware of the name of the
assay that contains the normalized expression values we want to provide as
input to GSVA. In this case this assay is called `logcounts`. Now we can
perform the first step to run GSVA, which consists of building a parameter
object.

```{r}
gsvapar <- gsvaParam(sce, genesets, assay="logcounts")
gsvapar
```
In the second and final step, we call the function `gsva()` with the previous
parameter object as input.

```{r}
es <- gsva(gsvapar)
```

# Using GSVA scores to assign cell types

Following [@amezquita2020clustering], and some of the steps described in Chapter
5 Clustering of the
[OSCA book](https://bioconductor.org/books/3.16/OSCA/clustering.html), here we
use GSVA scores to create first a nearst-neighbor graph of the cells using the
function `buildSNNGraph()` from the Bioconductor package
[scran](https://bioconductor.org/packages/scran), and then use this graph to
cluster the cells using the community detection algorithm implemented in the
function `cluster_walktrap()` from the CRAN package
[igraph](https://bioconductor.org/packages/igraph).

```{r, message=FALSE, warning=FALSE}
library(scran)
library(igraph)

g <- buildSNNGraph(es, k=20, assay.type="es")

colLabels(es) <- factor(cluster_walktrap(g)$membership)
table(colLabels(es))
```
We set `k=20` in the call to `buildSNNGraph()` so that we get a number of
clusters close to the number of cell types found in PBMCs (T cells, B cells,
NK cells, monocytes and dendritic cells).

Similarly to @diaz2019evaluation, we apply a very simple cell type assignment
algorithm, which consists of selecting at each cell the gene set with highest
GSVA score, tallying the selected gene sets per cluster, and assigning to the
cluster the most frequent gene set, storing tat assignment in the column data
of the `SingleCellExperiment` object.

```{r}
whmax <- apply(assay(es), 2, which.max)
gsetsxlabels <- split(rownames(es)[whmax], colLabels(es))
gsetsxlabels <- names(sapply(sapply(gsetsxlabels, table), which.max))
colData(es)$gsetsxlabel <- gsetsxlabels[es$label]
```
Finally, using the Bioconductor package
[scater](https://bioconductor.org/packages/scater), we calculate the first
two PCA components on the GSVA scores and use them for visualizing in Figure
\@ref(fig:gsvaclusters) the similarity between cells in terms of GSVA scores
project in two dimensions, the clusters based on the community detection
algorithm and the cell type assignments based on gene sets.

```{r gsvaclusters, echo=TRUE, small.mar=TRUE, fig.height=4, fig.width=4, units="in", dpi=100, fig.cap="Cell type assignments of PBMC scRNA-seq data, based on GSVA scores."}
library(scater)

set.seed(123)
es <- runPCA(es, assay.type="es", ncomponents=2)
plotReducedDim(es, dimred="PCA", colour_by="label", text_by="gsetsxlabel")
```

# Session information

```{r}
sessionInfo()
```

# References
