<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="pandoc">
<meta http-equiv="X-UA-Compatible" content="IE=EDGE">
<meta name="date" content="2024-08-31">
<title>Analysis of bulk RNA-seq data</title>
<script src="bulkRNAseq_files/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1">
<link href="bulkRNAseq_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet">
<script src="bulkRNAseq_files/bootstrap-3.3.5/js/bootstrap.min.js"></script><script src="bulkRNAseq_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script><script src="bulkRNAseq_files/bootstrap-3.3.5/shim/respond.min.js"></script><style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       </style>
<script src="bulkRNAseq_files/navigation-1.1/tabsets.js"></script><style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>
<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script><style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>
<link rel="stylesheet" href="/usr/local/lib/R/site-library/BiocStyle/resources/html/bioconductor.css" type="text/css">
<style type="text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
</style>
<!-- tabsets --><style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>
<!-- code folding --><script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>
</head>
<body>


<div class="container-fluid main-container">




<div id="header">



<h1 class="title toc-ignore">Analysis of bulk RNA-seq data</h1>
<p class="author-name">Axel Klenk<span class="affil-mark">1*</span> and Robert Castelo<span class="affil-mark">1**</span></p>
<p class="author-affiliation"><span class="affil-mark">1</span>Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona</p>
<p class="author-email"><span class="affil-mark">*</span><a href="mailto:axelvolker.klenk@upf.edu">axelvolker.klenk@upf.edu</a><br><span class="affil-mark">**</span><a href="mailto:robert.castelo@upf.edu">robert.castelo@upf.edu</a></p>
<h4 class="date">31 August 2024</h4>

</div>

<h1>Contents</h1>
<div id="TOC">
<ul>
<li><a href="#importing-bulk-rna-seq-data-into-a-summarizedexperiment-object" id="toc-importing-bulk-rna-seq-data-into-a-summarizedexperiment-object"><span class="toc-section-number">1</span> Importing bulk RNA-seq data into a <code>SummarizedExperiment</code> object</a></li>
<li><a href="#data-exploration-at-gene-level" id="toc-data-exploration-at-gene-level"><span class="toc-section-number">2</span> Data exploration at gene level</a></li>
<li><a href="#importing-gene-sets" id="toc-importing-gene-sets"><span class="toc-section-number">3</span> Importing gene sets</a></li>
<li><a href="#how-to-run-gsva" id="toc-how-to-run-gsva"><span class="toc-section-number">4</span> How to run GSVA</a></li>
<li><a href="#data-exploration-at-pathway-level" id="toc-data-exploration-at-pathway-level"><span class="toc-section-number">5</span> Data exploration at pathway level</a></li>
<li><a href="#differential-expression-at-pathway-level" id="toc-differential-expression-at-pathway-level"><span class="toc-section-number">6</span> Differential expression at pathway level</a></li>
<li><a href="#session-information" id="toc-session-information"><span class="toc-section-number">7</span> Session information</a></li>
<li><a href="#references" id="toc-references">References</a></li>
</ul>
</div>

<div class="section level2" number="1">
<h2 id="importing-bulk-rna-seq-data-into-a-summarizedexperiment-object">
<span class="header-section-number">1</span> Importing bulk RNA-seq data into a <code>SummarizedExperiment</code> object<a class="anchor" aria-label="anchor" href="#importing-bulk-rna-seq-data-into-a-summarizedexperiment-object"></a>
</h2>
<p>We will illustrate how to use GSVA with the bulk RNA-seq data from
<span class="citation">(<a href="#ref-costa2021genome">Costa et al. 2021</a>)</span>. This dataset consists of stranded 2x75nt paired-end
reads sequenced from whole blood stored in
<a href="https://en.wikipedia.org/wiki/Dried_blood_spot" class="external-link">dried blood spots (DBS)</a>.
<span class="citation">Costa et al. (<a href="#ref-costa2021genome">2021</a>)</span> generated these data from 21 DBS samples of extremely
preterm newborns (neonates born before the 28th week of gestation), where 10 of
them had been exposed to a fetal inflammatory response (FIR) before birth.</p>
<p>We start by building a
<a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a>
object from these RNA-seq data. A corresponding matrix of normalized log-CPM
units of expression is stored as part of this workshop package, where
lowly-expressed genes have been already filtered out. We should load these data
as follows.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fname</span> <span class="op">&lt;-</span> <span class="st">"CostaEtAl2021_FIRDBS_bulkRNAseq_data_ENTREZID.csv.gz"</span></span>
<span><span class="va">fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">fname</span><span class="op">)</span>,</span>
<span>                     package<span class="op">=</span><span class="st">"GSVAEuroBioC2024"</span><span class="op">)</span></span>
<span><span class="va">expr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">fname</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 11279    21</span></span>
<span><span class="va">expr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></span>
<span><span class="co">#&gt;               BS03     BS04     BS05     BS06     BS07     BS08     BS09</span></span>
<span><span class="co">#&gt; 100       1.907142 2.378443 1.416999 2.688953 2.541277 2.807271 3.299930</span></span>
<span><span class="co">#&gt; 10000     4.669820 4.492745 4.380702 5.134757 4.893896 4.408717 4.921101</span></span>
<span><span class="co">#&gt; 100008587 6.388405 7.167047 4.472078 6.851124 6.868794 4.531868 5.141607</span></span>
<span><span class="co">#&gt; 100009676 1.849203 1.467252 1.928039 1.266955 2.033022 1.369181 2.742960</span></span>
<span><span class="co">#&gt; 10001     4.463245 4.288214 4.075431 4.294054 4.220348 4.095428 4.033399</span></span></code></pre></div>
<p>Note that gene (row) identifiers correspond to
<a href="https://www.ncbi.nlm.nih.gov/gene" class="external-link">NCBI Entrez gene identifiers</a>, while sample
(column) identifiers correspond to anonymized neonates. This workshop package
also stores phenotype data describing the exposure to FIR and the sex of the
neonate.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fname</span> <span class="op">&lt;-</span> <span class="st">"CostaEtAl2021_FIRDBS_phenotype_data.csv.gz"</span></span>
<span><span class="va">fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">fname</span><span class="op">)</span>,</span>
<span>                     package<span class="op">=</span><span class="st">"GSVAEuroBioC2024"</span><span class="op">)</span></span>
<span><span class="va">pdat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">fname</span>, row.names<span class="op">=</span><span class="fl">1</span>, stringsAsFactors<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">pdat</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 21  2</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pdat</span><span class="op">)</span></span>
<span><span class="co">#&gt;      FIR    Sex</span></span>
<span><span class="co">#&gt; BS03 yes female</span></span>
<span><span class="co">#&gt; BS04 yes female</span></span>
<span><span class="co">#&gt; BS05 yes   male</span></span>
<span><span class="co">#&gt; BS06  no female</span></span>
<span><span class="co">#&gt; BS07  no female</span></span>
<span><span class="co">#&gt; BS08 yes   male</span></span></code></pre></div>
<p>We can observe that we have expression profiles for all four possible
combinations of FIR exposure and sex.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">pdat</span><span class="op">$</span><span class="va">FIR</span>, <span class="va">pdat</span><span class="op">$</span><span class="va">Sex</span><span class="op">)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;       female male</span></span>
<span><span class="co">#&gt;   no       4    7</span></span>
<span><span class="co">#&gt;   yes      4    6</span></span></code></pre></div>
<p>We finally build the <code>SummarizedExperiment</code> object using the
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment()</a></code> constructor function from the
<a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a>
package. Prior to that call, we make sure that newborn identifiers in
the columns of the expression data are identical to the row identifiers
in the phenotype data. If this were not the case, then we must have made
some mistake when loading the CSV files into the <code>data.frame</code> objects
<code>expr</code> and <code>pdat</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## QC, make sure this line gives no error</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pdat</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span>assay<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>logCPM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                           colData<span class="op">=</span><span class="va">pdat</span><span class="op">)</span></span>
<span><span class="va">se</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 11279 21 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): logCPM</span></span>
<span><span class="co">#&gt; rownames(11279): 100 10000 ... 9994 9997</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(21): BS03 BS04 ... BS23 BS24</span></span>
<span><span class="co">#&gt; colData names(2): FIR Sex</span></span></code></pre></div>
<p>To facilitate later on the automatic mapping of gene identifiers between
gene sets and RNA-seq data, we should add annotation metadata to the
<code>SummarizedExperiment</code> object as follows.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rcastelo/GSVA" class="external-link">GSVA</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/gsvaAnnotation.html" class="external-link">gsvaAnnotation</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">EntrezIdentifier</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="2">
<h2 id="data-exploration-at-gene-level">
<span class="header-section-number">2</span> Data exploration at gene level<a class="anchor" aria-label="anchor" href="#data-exploration-at-gene-level"></a>
</h2>
<p>We do a brief data exploration at gene level, to have a sense of what can
we expect in our analysis at pathway level. Figure <a href="#fig:genelevelmds">1</a>
below shows the projection in two dimensions of sample dissmilarity by means
of a
<a href="https://en.wikipedia.org/wiki/Multidimensional_scaling" class="external-link">multidimensional scaling (MDS)</a>
plot, produced with the <code><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">plotMDS()</a></code> function of the Bioconductor package
<a href="https://bioconductor.org/packages/limma" class="external-link">limma</a>. We can observe that sample
dissimilarity in RNA expression from DBS samples is driven by the FIR and sex
phenotypes, as shown in Fig. 1C of <span class="citation">Costa et al. (<a href="#ref-costa2021genome">2021</a>)</span>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">limma</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">fircolor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>no<span class="op">=</span><span class="st">"skyblue"</span>, yes<span class="op">=</span><span class="st">"darkred"</span><span class="op">)</span></span>
<span><span class="va">sexpch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>female<span class="op">=</span><span class="fl">19</span>, male<span class="op">=</span><span class="fl">15</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">plotMDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span>, col<span class="op">=</span><span class="va">fircolor</span><span class="op">[</span><span class="va">se</span><span class="op">$</span><span class="va">FIR</span><span class="op">]</span>, pch<span class="op">=</span><span class="va">sexpch</span><span class="op">[</span><span class="va">se</span><span class="op">$</span><span class="va">Sex</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:genelevelmds"></span>
<img src="bulkRNAseq_files/figure-html/genelevelmds-1.png" alt="Gene-level exploration. Multidimensional scaling (MDS) plot at gene level. Red corresponds to `FIR=yes` and blue to `FIR=no`, while circles and squares correspond, respectively, to female and male neonates." width="100%" class="widefigure"><p class="caption">
Figure 1: <span class="caption-title">Gene-level exploration</span><br>Multidimensional scaling (MDS) plot at gene level. Red corresponds to <code>FIR=yes</code> and blue to <code>FIR=no</code>, while circles and squares correspond, respectively, to female and male neonates.
</p>
</div>
</div>
<div class="section level2" number="3">
<h2 id="importing-gene-sets">
<span class="header-section-number">3</span> Importing gene sets<a class="anchor" aria-label="anchor" href="#importing-gene-sets"></a>
</h2>
<p><span class="citation">(<a href="#ref-Costa2021"><strong>Costa2021?</strong></a>)</span> genome report a postnatal activation of the innate immune system
and an impairment of the adaptive immunity. For the purpose of exploring these
results at pathway level, we will download the C7 collection of immunologic
signature gene sets from
<a href="https://www.gsea-msigdb.org/gsea/msigdb/human/collection_details.jsp#C7" class="external-link">MSigDB</a>,
concretely the subcollection ImmuneSigDB. Here we use the GSVA function
<code><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT()</a></code>, which allow us to directly download the GMT file from the MSigDB
database.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">URL</span> <span class="op">&lt;-</span> <span class="st">"https://data.broadinstitute.org/gsea-msigdb/msigdb/release/2023.2.Hs/c7.immunesigdb.v2023.2.Hs.symbols.gmt"</span></span>
<span><span class="va">genesets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT</a></span><span class="op">(</span><span class="va">URL</span>, geneIdType<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>However, for the purpose of this workshop, if we have problems accessing that
URL, we can also load a compressed version stored in the workshop package, as
follows.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"c7.immunesigdb.v2023.2.Hs.symbols.gmt.gz"</span>,</span>
<span>                     package<span class="op">=</span><span class="st">"GSVAEuroBioC2024"</span><span class="op">)</span></span>
<span><span class="va">genesets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT</a></span><span class="op">(</span><span class="va">fname</span>, geneIdType<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">genesets</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: GOLDRATH_EFF_VS_MEMORY_CD8_TCELL_DN, GOLDRATH_EFF_VS_MEMORY_CD8_TCELL_UP, ..., KAECH_NAIVE_VS_MEMORY_CD8_TCELL_UP (4872 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: ABCA2, ABCC5, ..., LINC00841 (20457 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: NullCollection (1 total)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 4872</span></span></code></pre></div>
<p>The argument <code>geneIdType=SymbolIdentifier()</code> adds metadata so that the
resulting <code>GeneSetCollection</code> object knows that the gene identifiers employed
in this gene set collection correspond to
<a href="https://www.genenames.org" class="external-link">HUGO symbols</a>.</p>
<p>To reduce the execution time in this workshop, we are going to select a subset
of gene sets that are more relevant to our data. Concretely, we select gene
sets formed by genes upregulated in innate leukocytes and adaptive mature
lymphocytes, excluding those reported in studies on myeloid cells and the lupus
autoimmune disease.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">innatepat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NKCELL_VS_.+_UP"</span>, <span class="st">"MAST_CELL_VS_.+_UP"</span>,</span>
<span>               <span class="st">"EOSINOPHIL_VS_.+_UP"</span>, <span class="st">"BASOPHIL_VS_.+_UP"</span>,</span>
<span>               <span class="st">"MACROPHAGE_VS_.+_UP"</span>, <span class="st">"NEUTROPHIL_VS_.+_UP"</span><span class="op">)</span></span>
<span><span class="va">innatepat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">innatepat</span>, collapse<span class="op">=</span><span class="st">"|"</span><span class="op">)</span></span>
<span><span class="va">innategsets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="va">innatepat</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">innategsets</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 53</span></span>
<span></span>
<span><span class="va">adaptivepat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD4_TCELL_VS_.+_UP"</span>, <span class="st">"CD8_TCELL_VS_.+_UP"</span>, <span class="st">"BCELL_VS_.+_UP"</span><span class="op">)</span></span>
<span><span class="va">adaptivepat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">adaptivepat</span>, collapse<span class="op">=</span><span class="st">"|"</span><span class="op">)</span></span>
<span><span class="va">adaptivegsets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="va">adaptivepat</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">excludepat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NAIVE"</span>, <span class="st">"LUPUS"</span>, <span class="st">"MYELOID"</span><span class="op">)</span></span>
<span><span class="va">excludepat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">excludepat</span>, collapse<span class="op">=</span><span class="st">"|"</span><span class="op">)</span></span>
<span><span class="va">adaptivegsets</span> <span class="op">&lt;-</span> <span class="va">adaptivegsets</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="va">excludepat</span>, <span class="va">adaptivegsets</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">adaptivegsets</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 97</span></span>
<span></span>
<span><span class="va">genesets</span> <span class="op">&lt;-</span> <span class="va">genesets</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">innategsets</span>, <span class="va">adaptivegsets</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 150</span></span></code></pre></div>
</div>
<div class="section level2" number="4">
<h2 id="how-to-run-gsva">
<span class="header-section-number">4</span> How to run GSVA<a class="anchor" aria-label="anchor" href="#how-to-run-gsva"></a>
</h2>
<p>A GSVA analysis has two main inputs, the molecular data (e.g., gene expression),
and a collection of gene sets, and we run it in two steps. First, we should
build a parameter object for the desired methodology. Here we use the GSVA
algorithm of <span class="citation">Hänzelmann, Castelo, and Guinney (<a href="#ref-hanzelmann2013gsva">2013</a>)</span> by calling the function <code><a href="https://rdrr.io/pkg/GSVA/man/gsvaParam-class.html" class="external-link">gsvaParam()</a></code> (see the
<a href="https://bioconductor.org/packages/GSVA" class="external-link">GSVA</a> package vignette for other
functions implementing different pathway analysis methods).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsvapar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/gsvaParam-class.html" class="external-link">gsvaParam</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">genesets</span>, assay<span class="op">=</span><span class="st">"logCPM"</span>, minSize<span class="op">=</span><span class="fl">5</span>, maxSize<span class="op">=</span><span class="fl">300</span><span class="op">)</span></span>
<span><span class="va">gsvapar</span></span>
<span><span class="co">#&gt; A GSVA::gsvaParam object</span></span>
<span><span class="co">#&gt; expression data:</span></span>
<span><span class="co">#&gt;   class: SummarizedExperiment </span></span>
<span><span class="co">#&gt;   dim: 11279 21 </span></span>
<span><span class="co">#&gt;   metadata(1): annotation</span></span>
<span><span class="co">#&gt;   assays(1): logCPM</span></span>
<span><span class="co">#&gt;   rownames(11279): 100 10000 ... 9994 9997</span></span>
<span><span class="co">#&gt;   rowData names(0):</span></span>
<span><span class="co">#&gt;   colnames(21): BS03 BS04 ... BS23 BS24</span></span>
<span><span class="co">#&gt;   colData names(2): FIR Sex</span></span>
<span><span class="co">#&gt; using assay: logCPM</span></span>
<span><span class="co">#&gt; using annotation: none</span></span>
<span><span class="co">#&gt; gene sets:</span></span>
<span><span class="co">#&gt;   GeneSetCollection</span></span>
<span><span class="co">#&gt;     names: GSE18804_SPLEEN_MACROPHAGE_VS_BRAIN_TUMORAL_MACROPHAGE_UP, GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_UP, ..., GSE7460_CD8_TCELL_VS_TREG_ACT_UP (150 total)</span></span>
<span><span class="co">#&gt;     unique identifiers: ABCA8, ABLIM2, ..., ZFP30 (11353 total)</span></span>
<span><span class="co">#&gt;     types in collection:</span></span>
<span><span class="co">#&gt;       geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;       collectionType: NullCollection (1 total)</span></span>
<span><span class="co">#&gt; gene set size: [5, 300]</span></span>
<span><span class="co">#&gt; kcdf: auto</span></span>
<span><span class="co">#&gt; kcdfNoneMinSampleSize: 50</span></span>
<span><span class="co">#&gt; tau: 1</span></span>
<span><span class="co">#&gt; maxDiff: TRUE</span></span>
<span><span class="co">#&gt; absRanking: FALSE</span></span>
<span><span class="co">#&gt; sparse: TRUE</span></span></code></pre></div>
<p>In the second step, we call the <code><a href="https://rdrr.io/pkg/GSVA/man/gsva.html" class="external-link">gsva()</a></code> function with the parameter object
as its first argument, other arguments to this function control its verbosity
and parellel execution through the Bioconductor package
<a href="https://bioconductor.org/packages/BiocParallel" class="external-link">BiocParallel</a>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>es <span class="ot">&lt;-</span> <span class="fu">gsva</span>(gsvapar)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="co">#&gt; Mapping identifiers between gene sets and feature names.</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="co">#&gt; Estimating GSVA scores for 150 gene sets.</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co">#&gt; Figuring out automatically how to estimate ECDFs</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt; Estimating ECDFs with Gaussian kernels</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>  <span class="er">|===</span>                                                                   <span class="er">|</span>   <span class="dv">5</span>%</span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>  <span class="er">|=======</span>                                                               <span class="er">|</span>  <span class="dv">10</span>%</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a>  <span class="er">|==========</span>                                                            <span class="er">|</span>  <span class="dv">14</span>%</span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a>  <span class="er">|=============</span>                                                         <span class="er">|</span>  <span class="dv">19</span>%</span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a>  <span class="er">|=================</span>                                                     <span class="er">|</span>  <span class="dv">24</span>%</span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a>  <span class="er">|====================</span>                                                  <span class="er">|</span>  <span class="dv">29</span>%</span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-22"><a href="#cb11-22" tabindex="-1"></a>  <span class="er">|=======================</span>                                               <span class="er">|</span>  <span class="dv">33</span>%</span>
<span id="cb11-23"><a href="#cb11-23" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-24"><a href="#cb11-24" tabindex="-1"></a>  <span class="er">|===========================</span>                                           <span class="er">|</span>  <span class="dv">38</span>%</span>
<span id="cb11-25"><a href="#cb11-25" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-26"><a href="#cb11-26" tabindex="-1"></a>  <span class="er">|==============================</span>                                        <span class="er">|</span>  <span class="dv">43</span>%</span>
<span id="cb11-27"><a href="#cb11-27" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-28"><a href="#cb11-28" tabindex="-1"></a>  <span class="er">|=================================</span>                                     <span class="er">|</span>  <span class="dv">48</span>%</span>
<span id="cb11-29"><a href="#cb11-29" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-30"><a href="#cb11-30" tabindex="-1"></a>  <span class="er">|=====================================</span>                                 <span class="er">|</span>  <span class="dv">52</span>%</span>
<span id="cb11-31"><a href="#cb11-31" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-32"><a href="#cb11-32" tabindex="-1"></a>  <span class="er">|========================================</span>                              <span class="er">|</span>  <span class="dv">57</span>%</span>
<span id="cb11-33"><a href="#cb11-33" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-34"><a href="#cb11-34" tabindex="-1"></a>  <span class="er">|===========================================</span>                           <span class="er">|</span>  <span class="dv">62</span>%</span>
<span id="cb11-35"><a href="#cb11-35" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-36"><a href="#cb11-36" tabindex="-1"></a>  <span class="er">|===============================================</span>                       <span class="er">|</span>  <span class="dv">67</span>%</span>
<span id="cb11-37"><a href="#cb11-37" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-38"><a href="#cb11-38" tabindex="-1"></a>  <span class="er">|==================================================</span>                    <span class="er">|</span>  <span class="dv">71</span>%</span>
<span id="cb11-39"><a href="#cb11-39" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-40"><a href="#cb11-40" tabindex="-1"></a>  <span class="er">|=====================================================</span>                 <span class="er">|</span>  <span class="dv">76</span>%</span>
<span id="cb11-41"><a href="#cb11-41" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-42"><a href="#cb11-42" tabindex="-1"></a>  <span class="er">|=========================================================</span>             <span class="er">|</span>  <span class="dv">81</span>%</span>
<span id="cb11-43"><a href="#cb11-43" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-44"><a href="#cb11-44" tabindex="-1"></a>  <span class="er">|============================================================</span>          <span class="er">|</span>  <span class="dv">86</span>%</span>
<span id="cb11-45"><a href="#cb11-45" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-46"><a href="#cb11-46" tabindex="-1"></a>  <span class="er">|===============================================================</span>       <span class="er">|</span>  <span class="dv">90</span>%</span>
<span id="cb11-47"><a href="#cb11-47" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-48"><a href="#cb11-48" tabindex="-1"></a>  <span class="er">|===================================================================</span>   <span class="er">|</span>  <span class="dv">95</span>%</span>
<span id="cb11-49"><a href="#cb11-49" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb11-50"><a href="#cb11-50" tabindex="-1"></a>  <span class="er">|======================================================================|</span> <span class="dv">100</span>%</span>
<span id="cb11-51"><a href="#cb11-51" tabindex="-1"></a>es</span>
<span id="cb11-52"><a href="#cb11-52" tabindex="-1"></a><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span id="cb11-53"><a href="#cb11-53" tabindex="-1"></a><span class="co">#&gt; dim: 150 21 </span></span>
<span id="cb11-54"><a href="#cb11-54" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb11-55"><a href="#cb11-55" tabindex="-1"></a><span class="co">#&gt; assays(1): es</span></span>
<span id="cb11-56"><a href="#cb11-56" tabindex="-1"></a><span class="co">#&gt; rownames(150):</span></span>
<span id="cb11-57"><a href="#cb11-57" tabindex="-1"></a><span class="co">#&gt;   GSE18804_SPLEEN_MACROPHAGE_VS_BRAIN_TUMORAL_MACROPHAGE_UP</span></span>
<span id="cb11-58"><a href="#cb11-58" tabindex="-1"></a><span class="co">#&gt;   GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_UP ...</span></span>
<span id="cb11-59"><a href="#cb11-59" tabindex="-1"></a><span class="co">#&gt;   GSE7460_CD8_TCELL_VS_CD4_TCELL_ACT_UP</span></span>
<span id="cb11-60"><a href="#cb11-60" tabindex="-1"></a><span class="co">#&gt;   GSE7460_CD8_TCELL_VS_TREG_ACT_UP</span></span>
<span id="cb11-61"><a href="#cb11-61" tabindex="-1"></a><span class="co">#&gt; rowData names(1): gs</span></span>
<span id="cb11-62"><a href="#cb11-62" tabindex="-1"></a><span class="co">#&gt; colnames(21): BS03 BS04 ... BS23 BS24</span></span>
<span id="cb11-63"><a href="#cb11-63" tabindex="-1"></a><span class="co">#&gt; colData names(2): FIR Sex</span></span></code></pre></div>
<p>Because the input expression data was provided in a <code>SummmarizedExperiment</code>
object, the output of <code><a href="https://rdrr.io/pkg/GSVA/man/gsva.html" class="external-link">gsva()</a></code> is an identical <code>SummarizedExperiment</code> object,
with the exception that the one or more matrices of molecular data in the
assay slot of the object have been replaced by a single matrix of GSVA
enrichment scores.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "logCPM"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "es"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;                                                                  BS03</span></span>
<span><span class="co">#&gt; GSE18804_SPLEEN_MACROPHAGE_VS_BRAIN_TUMORAL_MACROPHAGE_UP -0.06952401</span></span>
<span><span class="co">#&gt; GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_UP  0.28088461</span></span>
<span><span class="co">#&gt; GSE18804_SPLEEN_MACROPHAGE_VS_TUMORAL_MACROPHAGE_UP       -0.13659475</span></span>
<span><span class="co">#&gt;                                                                 BS04</span></span>
<span><span class="co">#&gt; GSE18804_SPLEEN_MACROPHAGE_VS_BRAIN_TUMORAL_MACROPHAGE_UP  0.0530275</span></span>
<span><span class="co">#&gt; GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_UP -0.1305626</span></span>
<span><span class="co">#&gt; GSE18804_SPLEEN_MACROPHAGE_VS_TUMORAL_MACROPHAGE_UP        0.1634395</span></span>
<span><span class="co">#&gt;                                                                  BS05</span></span>
<span><span class="co">#&gt; GSE18804_SPLEEN_MACROPHAGE_VS_BRAIN_TUMORAL_MACROPHAGE_UP -0.13344892</span></span>
<span><span class="co">#&gt; GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_UP -0.08684864</span></span>
<span><span class="co">#&gt; GSE18804_SPLEEN_MACROPHAGE_VS_TUMORAL_MACROPHAGE_UP       -0.17536163</span></span></code></pre></div>
</div>
<div class="section level2" number="5">
<h2 id="data-exploration-at-pathway-level">
<span class="header-section-number">5</span> Data exploration at pathway level<a class="anchor" aria-label="anchor" href="#data-exploration-at-pathway-level"></a>
</h2>
<p>We do again a data exploration, this time at pathway level. Figure
<a href="#fig:pathwaylevelmds">2</a> below, shows an MDS plot of GSVA enrichment scores.
We can see again that most variability is driven by the FIR phenotype, but this
time the sex phenotype does not seem to affect sample dissimilarity at pathway
level, probably because the collection of gene sets we have used does not
include gene sets formed by genes with sex-specific expression.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">plotMDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span>, col<span class="op">=</span><span class="va">fircolor</span><span class="op">[</span><span class="va">es</span><span class="op">$</span><span class="va">FIR</span><span class="op">]</span>, pch<span class="op">=</span><span class="va">sexpch</span><span class="op">[</span><span class="va">es</span><span class="op">$</span><span class="va">Sex</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:pathwaylevelmds"></span>
<img src="bulkRNAseq_files/figure-html/pathwaylevelmds-1.png" alt="Pathway-level exploration. Multidimensional scaling (MDS) plot at pathway level. Red corresponds to `FIR=yes` and blue to `FIR=no`, while circles and squares correspond, respectively, to female and male neonates." width="100%" class="widefigure"><p class="caption">
Figure 2: <span class="caption-title">Pathway-level exploration</span><br>Multidimensional scaling (MDS) plot at pathway level. Red corresponds to <code>FIR=yes</code> and blue to <code>FIR=no</code>, while circles and squares correspond, respectively, to female and male neonates.
</p>
</div>
</div>
<div class="section level2" number="6">
<h2 id="differential-expression-at-pathway-level">
<span class="header-section-number">6</span> Differential expression at pathway level<a class="anchor" aria-label="anchor" href="#differential-expression-at-pathway-level"></a>
</h2>
<p>One of the most common pathway-centric analyses is a differential expression
analysis at pathway level. We conduct such an analysis using the Bioconductor
packages <a href="https://bioconductor.org/packages/limma" class="external-link">limma</a> and
<a href="https://bioconductor.org/packages/sva" class="external-link">sva</a>, the latter to adjust for sample
heterogeneity using surrogate variable analysis <span class="citation">(<a href="#ref-leek2007capturing">Leek and Storey 2007</a>)</span>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">sva</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">limma</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## build design matrix of the model to which we fit the data</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">FIR</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## build design matrix of the corresponding null model</span></span>
<span><span class="va">mod0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## estimate surrogate variables (SVs) with SVA</span></span>
<span><span class="va">sv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sva/man/sva.html" class="external-link">sva</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span>, <span class="va">mod</span>, <span class="va">mod0</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of significant surrogate variables is:  4 </span></span>
<span><span class="co">#&gt; Iteration (out of 5 ):1  2  3  4  5</span></span>
<span><span class="co">## add SVs to the design matrix of the model of interest</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">mod</span>, <span class="va">sv</span><span class="op">$</span><span class="va">sv</span><span class="op">)</span></span>
<span><span class="co">## fit linear models</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span>, <span class="va">mod</span><span class="op">)</span></span>
<span><span class="co">## calculate moderated t-statistics using the robust regime</span></span>
<span><span class="va">fit.eb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span>
<span><span class="co">## summarize the extent of differential expression at 5% FDR</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/decideTests.html" class="external-link">decideTests</a></span><span class="op">(</span><span class="va">fit.eb</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;        (Intercept) FIRyes                </span></span>
<span><span class="co">#&gt; Down            24     12  38  40  33  54</span></span>
<span><span class="co">#&gt; NotSig         118     99  65  78  70  93</span></span>
<span><span class="co">#&gt; Up               8     39  47  32  47   3</span></span></code></pre></div>
<p>As shown in Figure <a href="#fig:esstdevxgssize">3</a> below, we may observe that GSVA
scores have higher precision for larger gene sets. GSVA scores derived from
gene sets that include small sizes (our smallest gene set here is about
100 genes), and from less heterogenous expression data, show such a trend
more clearly.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gssizes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/geneSets.html" class="external-link">geneSetSizes</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></span><span class="op">(</span><span class="va">gssizes</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></span><span class="op">(</span><span class="va">fit.eb</span><span class="op">$</span><span class="va">sigma</span><span class="op">)</span>, xlab<span class="op">=</span><span class="st">"Sqrt(gene sets sizes)"</span>,</span>
<span>          ylab<span class="op">=</span><span class="st">"Sqrt(standard deviation)"</span>, las<span class="op">=</span><span class="fl">1</span>, pch<span class="op">=</span><span class="st">"."</span>, cex<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/lowess.html" class="external-link">lowess</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></span><span class="op">(</span><span class="va">gssizes</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></span><span class="op">(</span><span class="va">fit.eb</span><span class="op">$</span><span class="va">sigma</span><span class="op">)</span><span class="op">)</span>, col<span class="op">=</span><span class="st">"red"</span>, lwd<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:esstdevxgssize"></span>
<img src="bulkRNAseq_files/figure-html/esstdevxgssize-1.png" alt="Pathway-level differential expression analysis. Residual standard deviation of GSVA scores as a function of gene set size. Larger gene sets tend to have higher precision." width="100%" class="widefigure"><p class="caption">
Figure 3: <span class="caption-title">Pathway-level differential expression analysis</span><br>Residual standard deviation of GSVA scores as a function of gene set size. Larger gene sets tend to have higher precision.
</p>
</div>
<p>When this trend is present, we can use improve the statistical power to detect
differentially expressed (DE) pathways by using the limma-trend pipeline. More
concretely, we should call the <code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes()</a></code> function with the argument <code>trend=x</code>,
where <code>x</code> is a vector of values corresponding to the sizes of the gene sets.
As we can see in the previous code chunk, these size values can be easily
obtained using GSVA’s function <code><a href="https://rdrr.io/pkg/GSVA/man/geneSets.html" class="external-link">geneSetSizes()</a></code> on the output of the <code><a href="https://rdrr.io/pkg/GSVA/man/gsva.html" class="external-link">gsva()</a></code>
function. Here below, we call again <code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes()</a></code> using the <code>trend</code> parameter. In
this case, however, the change in the number of DE pathway is negligible.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit.eb.trend</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span>, trend<span class="op">=</span><span class="va">gssizes</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/decideTests.html" class="external-link">decideTests</a></span><span class="op">(</span><span class="va">fit.eb.trend</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;        (Intercept) FIRyes                </span></span>
<span><span class="co">#&gt; Down            24     12  39  41  34  59</span></span>
<span><span class="co">#&gt; NotSig         121     99  64  76  69  88</span></span>
<span><span class="co">#&gt; Up               5     39  47  33  47   3</span></span></code></pre></div>
<p>We can select DE pathways with FDR &lt; 5% as follows.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fit.eb.trend</span>, coef<span class="op">=</span><span class="fl">2</span>, n<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span></span>
<span><span class="va">DEpwys</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tt</span><span class="op">)</span><span class="op">[</span><span class="va">tt</span><span class="op">$</span><span class="va">adj.P.Val</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">DEpwys</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 51</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">DEpwys</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "GSE3039_CD4_TCELL_VS_ALPHABETA_CD8_TCELL_UP"           </span></span>
<span><span class="co">#&gt; [2] "GSE3982_EOSINOPHIL_VS_CENT_MEMORY_CD4_TCELL_UP"        </span></span>
<span><span class="co">#&gt; [3] "GSE3982_NEUTROPHIL_VS_CENT_MEMORY_CD4_TCELL_UP"        </span></span>
<span><span class="co">#&gt; [4] "GSE36392_EOSINOPHIL_VS_NEUTROPHIL_IL25_TREATED_LUNG_UP"</span></span>
<span><span class="co">#&gt; [5] "GSE3982_NEUTROPHIL_VS_TH2_UP"                          </span></span>
<span><span class="co">#&gt; [6] "GSE22886_NEUTROPHIL_VS_DC_UP"</span></span></code></pre></div>
<p>Figure <a href="#fig:heatmapdepwys">4</a> below shows a heatmap of the GSVA enrichment
scores of the subset of the 51 DE pathways, clustered by
pathway and sample. We may observe that, consistently with the findings of
<span class="citation">Costa et al. (<a href="#ref-costa2021genome">2021</a>)</span>, FIR-affected neonates display an enrichment of upregulated
pathways associated with innate immunity, and an enrichment of downregulated
pathways associated with adaptive immunity, with respect to
FIR-unaffected neonates.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DEpwys_es</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">es</span><span class="op">[</span><span class="va">DEpwys</span>, <span class="op">]</span><span class="op">)</span></span>
<span><span class="va">sam_col_map</span> <span class="op">&lt;-</span> <span class="va">fircolor</span><span class="op">[</span><span class="va">es</span><span class="op">$</span><span class="va">FIR</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sam_col_map</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">DEpwys_es</span><span class="op">)</span></span>
<span><span class="va">sampleClust</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="fl">1</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">DEpwys_es</span>, method<span class="op">=</span><span class="st">"spearman"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                      method<span class="op">=</span><span class="st">"complete"</span><span class="op">)</span></span>
<span><span class="va">gsetClust</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="fl">1</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">DEpwys_es</span><span class="op">)</span>, method<span class="op">=</span><span class="st">"pearson"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                    method<span class="op">=</span><span class="st">"complete"</span><span class="op">)</span></span>
<span><span class="va">labrow</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">DEpwys_es</span><span class="op">)</span></span>
<span><span class="va">mask</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">DEpwys_es</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">innategsets</span></span>
<span><span class="va">labrow</span><span class="op">[</span><span class="va">mask</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"(INNATE)"</span>, <span class="va">labrow</span><span class="op">[</span><span class="va">mask</span><span class="op">]</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span></span>
<span><span class="va">mask</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">DEpwys_es</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">adaptivegsets</span></span>
<span><span class="va">labrow</span><span class="op">[</span><span class="va">mask</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"(ADAPTIVE)"</span>, <span class="va">labrow</span><span class="op">[</span><span class="va">mask</span><span class="op">]</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span></span>
<span><span class="va">labrow</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"_"</span>, <span class="st">" "</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"GSE[0-9]+_"</span>, <span class="st">""</span>, <span class="va">labrow</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/heatmap.html" class="external-link">heatmap</a></span><span class="op">(</span><span class="va">DEpwys_es</span>, ColSideColors<span class="op">=</span><span class="va">fircolor</span><span class="op">[</span><span class="va">es</span><span class="op">$</span><span class="va">FIR</span><span class="op">]</span>, xlab<span class="op">=</span><span class="st">"Samples"</span>,</span>
<span>        ylab<span class="op">=</span><span class="st">"Pathways"</span>, margins<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">20</span><span class="op">)</span>, labCol<span class="op">=</span><span class="st">""</span>, labRow<span class="op">=</span><span class="va">labrow</span>,</span>
<span>        scale<span class="op">=</span><span class="st">"row"</span>, Colv<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="va">sampleClust</span><span class="op">)</span>,</span>
<span>        Rowv<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="va">gsetClust</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:heatmapdepwys"></span>
<img src="bulkRNAseq_files/figure-html/heatmapdepwys-1.png" alt="Pathway-level signature of FIR. Heatmap of GSVA enrichment scores from pathways being called DE with 5% FDR between FIR-affected and unaffected neonates." width="100%" class="widefigure"><p class="caption">
Figure 4: <span class="caption-title">Pathway-level signature of FIR</span><br>Heatmap of GSVA enrichment scores from pathways being called DE with 5% FDR between FIR-affected and unaffected neonates.
</p>
</div>
</div>
<div class="section level2" number="7">
<h2 id="session-information">
<span class="header-section-number">7</span> Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] sva_3.53.0                  BiocParallel_1.39.0        </span></span>
<span><span class="co">#&gt;  [3] genefilter_1.87.0           mgcv_1.9-1                 </span></span>
<span><span class="co">#&gt;  [5] nlme_3.1-166                org.Hs.eg.db_3.19.1        </span></span>
<span><span class="co">#&gt;  [7] limma_3.61.9                GSVA_1.53.16               </span></span>
<span><span class="co">#&gt;  [9] GSEABase_1.67.0             graph_1.83.0               </span></span>
<span><span class="co">#&gt; [11] annotate_1.83.0             XML_3.99-0.17              </span></span>
<span><span class="co">#&gt; [13] AnnotationDbi_1.67.0        SummarizedExperiment_1.35.1</span></span>
<span><span class="co">#&gt; [15] Biobase_2.65.1              GenomicRanges_1.57.1       </span></span>
<span><span class="co">#&gt; [17] GenomeInfoDb_1.41.1         IRanges_2.39.2             </span></span>
<span><span class="co">#&gt; [19] S4Vectors_0.43.2            BiocGenerics_0.51.0        </span></span>
<span><span class="co">#&gt; [21] MatrixGenerics_1.17.0       matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt; [23] BiocStyle_2.33.1           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] blob_1.2.4                  Biostrings_2.73.1          </span></span>
<span><span class="co">#&gt;  [3] fastmap_1.2.0               SingleCellExperiment_1.27.2</span></span>
<span><span class="co">#&gt;  [5] digest_0.6.37               rsvd_1.0.5                 </span></span>
<span><span class="co">#&gt;  [7] lifecycle_1.0.4             survival_3.7-0             </span></span>
<span><span class="co">#&gt;  [9] statmod_1.5.0               KEGGREST_1.45.1            </span></span>
<span><span class="co">#&gt; [11] RSQLite_2.3.7               magrittr_2.0.3             </span></span>
<span><span class="co">#&gt; [13] compiler_4.4.1              rlang_1.1.4                </span></span>
<span><span class="co">#&gt; [15] sass_0.4.9                  tools_4.4.1                </span></span>
<span><span class="co">#&gt; [17] yaml_2.3.10                 knitr_1.48                 </span></span>
<span><span class="co">#&gt; [19] S4Arrays_1.5.7              htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt; [21] bit_4.0.5                   DelayedArray_0.31.11       </span></span>
<span><span class="co">#&gt; [23] abind_1.4-5                 HDF5Array_1.33.6           </span></span>
<span><span class="co">#&gt; [25] grid_4.4.1                  beachmat_2.21.5            </span></span>
<span><span class="co">#&gt; [27] xtable_1.8-4                edgeR_4.3.13               </span></span>
<span><span class="co">#&gt; [29] Rhdf5lib_1.27.0             tinytex_0.52               </span></span>
<span><span class="co">#&gt; [31] cli_3.6.3                   rmarkdown_2.28             </span></span>
<span><span class="co">#&gt; [33] crayon_1.5.3                httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [35] rjson_0.2.22                DBI_1.2.3                  </span></span>
<span><span class="co">#&gt; [37] cachem_1.1.0                rhdf5_2.49.0               </span></span>
<span><span class="co">#&gt; [39] splines_4.4.1               zlibbioc_1.51.1            </span></span>
<span><span class="co">#&gt; [41] parallel_4.4.1              BiocManager_1.30.25        </span></span>
<span><span class="co">#&gt; [43] XVector_0.45.0              vctrs_0.6.5                </span></span>
<span><span class="co">#&gt; [45] Matrix_1.7-0                jsonlite_1.8.8             </span></span>
<span><span class="co">#&gt; [47] bookdown_0.40               BiocSingular_1.21.2        </span></span>
<span><span class="co">#&gt; [49] bit64_4.0.5                 irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt; [51] magick_2.8.4                locfit_1.5-9.10            </span></span>
<span><span class="co">#&gt; [53] jquerylib_0.1.4             codetools_0.2-20           </span></span>
<span><span class="co">#&gt; [55] UCSC.utils_1.1.0            ScaledMatrix_1.13.0        </span></span>
<span><span class="co">#&gt; [57] htmltools_0.5.8.1           rhdf5filters_1.17.0        </span></span>
<span><span class="co">#&gt; [59] GenomeInfoDbData_1.2.12     R6_2.5.1                   </span></span>
<span><span class="co">#&gt; [61] sparseMatrixStats_1.17.2    evaluate_0.24.0            </span></span>
<span><span class="co">#&gt; [63] lattice_0.22-6              highr_0.11                 </span></span>
<span><span class="co">#&gt; [65] png_0.1-8                   SpatialExperiment_1.15.1   </span></span>
<span><span class="co">#&gt; [67] memoise_2.0.1               bslib_0.8.0                </span></span>
<span><span class="co">#&gt; [69] Rcpp_1.0.13                 SparseArray_1.5.31         </span></span>
<span><span class="co">#&gt; [71] xfun_0.47                   pkgconfig_2.0.3</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-costa2021genome" class="csl-entry">
Costa, Daniel, Núria Bonet, Amanda Solé, José Manuel González de Aledo-Castillo, Eduard Sabidó, Ferran Casals, Carlota Rovira, et al. 2021. <span>“Genome-Wide Postnatal Changes in Immunity Following Fetal Inflammatory Response.”</span> <em>The FEBS Journal</em> 288 (7): 2311–31. <a href="https://doi.org/10.1111/febs.15578" class="external-link">https://doi.org/10.1111/febs.15578</a>.
</div>
<div id="ref-hanzelmann2013gsva" class="csl-entry">
Hänzelmann, Sonja, Robert Castelo, and Justin Guinney. 2013. <span>“GSVA: Gene Set Variation Analysis for Microarray and RNA-Seq Data.”</span> <em>BMC Bioinformatics</em> 14: 1–15. <a href="https://doi.org/10.1186/1471-2105-14-7" class="external-link">https://doi.org/10.1186/1471-2105-14-7</a>.
</div>
<div id="ref-leek2007capturing" class="csl-entry">
Leek, Jeffrey T, and John D Storey. 2007. <span>“Capturing Heterogeneity in Gene Expression Studies by Surrogate Variable Analysis.”</span> <em>PLoS Genetics</em> 3 (9): e161. <a href="https://doi.org/10.1371/journal.pgen.0030161" class="external-link">https://doi.org/10.1371/journal.pgen.0030161</a>.
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script><!-- tabsets --><script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script><!-- code folding --><script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script><script type="text/javascript">
$(document).ready(function ()  {
  
  // Map "enter" keypress to the same action as a cursor click
  function navigateLink(e) {
    if (e.key === "Enter") {
      $(this).trigger("click");
    }
  }

  var toc_items = document.querySelectorAll(".tocify-item");
  for (var i = 0; i < toc_items.length; i++) {
    // The link role tells screen readers this is for navigation
    toc_items.item(i).setAttribute("role", "link");
    // tabindex = 0 allows selection via keyboard tab presses
    toc_items.item(i).setAttribute("tabindex", "0");
    // Listen for "Enter" keypress when item is selected
    toc_items.item(i).addEventListener("keydown", navigateLink);
  }
});
</script>
</body>
</html>
