<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Import gene sets from GMT files • GSVAEuroBioC2024</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Import gene sets from GMT files">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSVAEuroBioC2024</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/importGMTfiles.html">Import gene sets from GMT files</a>
    </li>
    <li>
      <a href="../articles/bulkRNAseq.html">Analysis of bulk RNA-seq data</a>
    </li>
    <li>
      <a href="../articles/scRNAseq.html">Analysis of single-cell RNA-seq data</a>
    </li>
    <li>
      <a href="../articles/spatial.html">Analysis of spatial omics data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/functionalgenomics/GSVAEuroBioC2024" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Import gene sets from GMT files</h1>
                        <h4 data-toc-skip class="author">Axel Klenk</h4>
            <address class="author_afil">
      Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona<br><a class="author_email" href="mailto:#"></a><a href="mailto:axelvolker.klenk@upf.edu" class="email">axelvolker.klenk@upf.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Robert Castelo</h4>
            <address class="author_afil">
      Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona<br><a class="author_email" href="mailto:#"></a><a href="mailto:robert.castelo@upf.edu" class="email">robert.castelo@upf.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">2024-09-03</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/functionalgenomics/GSVAEuroBioC2024/blob/devel/vignettes/importGMTfiles.Rmd" class="external-link"><code>vignettes/importGMTfiles.Rmd</code></a></small>
      <div class="hidden name"><code>importGMTfiles.Rmd</code></div>

    </div>

    
    
<p>Vignette built using <code>GSVA</code>, version 1.53.17.</p>
<div class="section level2" number="1">
<h2 id="what-are-gmt-files">
<span class="header-section-number">1</span> What are GMT files<a class="anchor" aria-label="anchor" href="#what-are-gmt-files"></a>
</h2>
<p>An important source of gene sets is the
<a href="https://www.gsea-msigdb.org/gsea/msigdb" class="external-link">Molecular Signatures Database (MSigDB)</a>
<span class="citation">(<a href="#ref-subramanian2005gene">Subramanian et al. 2005</a>)</span>, which stores them in plain text files following the
so-called
<a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29" class="external-link"><em>gene matrix transposed</em> (GMT) format</a>.
In the GMT format, each line stores a gene set with the following values
separated by tabs:</p>
<ul>
<li>A unique gene set identifier.</li>
<li>A gene set description.</li>
<li>One or more gene identifiers.</li>
</ul>
<p>Because each different gene set may consist of a different number of genes, each
line in a GMT file may contain a different number of tab-separated values. This
means that the GMT format is not a tabular format, and therefore cannot be directly
read with base R functions such as <code><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table()</a></code> or <code><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv()</a></code>.</p>
</div>
<div class="section level2" number="2">
<h2 id="how-to-read-gmt-files">
<span class="header-section-number">2</span> How to read GMT files<a class="anchor" aria-label="anchor" href="#how-to-read-gmt-files"></a>
</h2>
<p>We need a specialized function to read GMT files. We can find such a function in the
<a href="https://bioconductor.org/packages/GSEABase" class="external-link">GSEABase</a> package with <code><a href="https://rdrr.io/pkg/GSEABase/man/getObjects.html" class="external-link">getGmt()</a></code>, or
in the <a href="https://bioconductor.org/packages/qusage" class="external-link">qusage</a> package with <code>read.gmt()</code>.</p>
<p>GSVA also provides such a function called <code><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT()</a></code>, which takes as first argument
the filename or URL of a, possibly compressed, GMT file. The call below illustrates
how to read a GMT file from MSigDB providing its URL. Note that we also load the
package
<a href="https://bioconductor.org/packages/GSEABase" class="external-link">GSEABase</a> because, by default, the value
returned by <code><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT()</a></code> is a <code>GeneSetCollection</code> object defined in that package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rcastelo/GSVA" class="external-link">GSVA</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">URL</span> <span class="op">&lt;-</span> <span class="st">"https://data.broadinstitute.org/gsea-msigdb/msigdb/release/2023.2.Hs/c7.immunesigdb.v2023.2.Hs.symbols.gmt"</span></span>
<span><span class="va">genesets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT</a></span><span class="op">(</span><span class="va">URL</span>, geneIdType<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "GeneSetCollection"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "GSEABase"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 4872</span></span>
<span><span class="va">genesets</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: GOLDRATH_EFF_VS_MEMORY_CD8_TCELL_DN, GOLDRATH_EFF_VS_MEMORY_CD8_TCELL_UP, ..., KAECH_NAIVE_VS_MEMORY_CD8_TCELL_UP (4872 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: ABCA2, ABCC5, ..., LINC00841 (20457 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: NullCollection (1 total)</span></span></code></pre></div>
<p>If we have problems accessing that URL, we can also load a compressed version of this
GMT file, stored in the workshop package, as follows.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"c7.immunesigdb.v2023.2.Hs.symbols.gmt.gz"</span>,</span>
<span>                     package<span class="op">=</span><span class="st">"GSVAEuroBioC2024"</span><span class="op">)</span></span>
<span><span class="va">genesets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT</a></span><span class="op">(</span><span class="va">fname</span>, geneIdType<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "GeneSetCollection"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "GSEABase"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 4872</span></span>
<span><span class="va">genesets</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: GOLDRATH_EFF_VS_MEMORY_CD8_TCELL_DN, GOLDRATH_EFF_VS_MEMORY_CD8_TCELL_UP, ..., KAECH_NAIVE_VS_MEMORY_CD8_TCELL_UP (4872 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: ABCA2, ABCC5, ..., LINC00841 (20457 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: NullCollection (1 total)</span></span></code></pre></div>
<p>Next to the filename of the GMT file, we provided the argument
<code>geneIdType=SymbolIdentifier()</code>. This argument adds the necessary metadata that
later on allows other software to figure out what kind of gene identifiers are
used in this collection of gene sets, to attempt mapping them to other type of
identifiers, if necessary. While this argument is optional, we should always
try to provide it.</p>
</div>
<div class="section level2" number="3">
<h2 id="dealing-with-duplicated-gene-set-names">
<span class="header-section-number">3</span> Dealing with duplicated gene set names<a class="anchor" aria-label="anchor" href="#dealing-with-duplicated-gene-set-names"></a>
</h2>
<p>The specification of the <a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29" class="external-link">GMT format</a>
establishes that duplicated gene set names are not allowed. For this reason,
the <code><a href="https://rdrr.io/pkg/GSEABase/man/getObjects.html" class="external-link">getGmt()</a></code> function from the
<a href="https://bioconductor.org/packages/GSEABase" class="external-link">GSEABase</a> package prompts an error
when duplicated gene names are found, while the <code>read.gmt()</code> function from the
<a href="https://bioconductor.org/packages/qusage" class="external-link">qusage</a> package silently accepts them
in a list with duplicated element names.</p>
<p>The GSVA <code><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT()</a></code> function deals with duplicated gene set names as follows.
By default, <code><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT()</a></code> warns the user about a duplicated gene set name and
keeps only the first occurrence of the duplicated gene set in the returned
object. We can illustrate this situation with a GMT file from the MSigDB
database that happens to have duplicated gene set names and which we have
stored as part of this workshop package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"c2.all.v7.5.symbols.gmt.gz"</span>,</span>
<span>                     package<span class="op">=</span><span class="st">"GSVAEuroBioC2024"</span><span class="op">)</span></span>
<span><span class="va">genesets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/getObjects.html" class="external-link">getGmt</a></span><span class="op">(</span><span class="va">fname</span>, geneIdType<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in validObject(.Object): invalid class "GeneSetCollection" object: each setName must be distinct</span></span></code></pre></div>
<p>We can see that <code><a href="https://rdrr.io/pkg/GSEABase/man/getObjects.html" class="external-link">getGmt()</a></code> prompts an error. We can see below that this does
not happen with <code><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT()</a></code> and that, by default, all but the first occurrence of the
duplicated gene set have been removed.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genesets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT</a></span><span class="op">(</span><span class="va">fname</span>, geneIdType<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in deduplicateGmtLines(lines, deduplUse): GMT contains duplicated gene</span></span>
<span><span class="co">#&gt; set names; deduplicated using method: first</span></span>
<span><span class="va">genesets</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: CORONEL_RFX7_DIRECT_TARGETS_UP, FOROUTAN_TGFB_EMT_UP, ..., SA_TRKA_RECEPTOR (6363 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: ABAT, DIP2A, ..., IGHG3 (21728 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: NullCollection (1 total)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span></code></pre></div>
<p>The parameter <code>deduplUse</code> in the <code><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT()</a></code> function allow one to apply other
policies to deal with duplicated gene set names, see its help page with
<code><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">?readGMT</a></code>.</p>
</div>
<div class="section level2" number="4">
<h2 id="returning-a-list-instead-of-a-genesetcollection">
<span class="header-section-number">4</span> Returning a <code>list</code> instead of a <code>GeneSetCollection</code><a class="anchor" aria-label="anchor" href="#returning-a-list-instead-of-a-genesetcollection"></a>
</h2>
<p>Storing gene sets in a <code>GeneSetCollection</code> object offers a number of advantages,
including the addition of metadata that enables an automatic mapping of the gene
identifiers that form the gene sets. For this reason, <code><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT()</a></code> returns such
an object by default. However, if a user prefers using gene sets stored as a
<code>list</code> object of <code>character</code> vectors, this is possible by setting the argument
<code>valueType="list"</code> in the call to <code><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genesets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT</a></span><span class="op">(</span><span class="va">fname</span>, valueType<span class="op">=</span><span class="st">"list"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in deduplicateGmtLines(lines, deduplUse): GMT contains duplicated gene</span></span>
<span><span class="co">#&gt; set names; deduplicated using method: first</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "list"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">genesets</span>, <span class="va">head</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; $CORONEL_RFX7_DIRECT_TARGETS_UP</span></span>
<span><span class="co">#&gt; [1] "ABAT"  "DIP2A" "KLF9"  "PTMS"  "TOB2"  "ARL15"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $FOROUTAN_TGFB_EMT_UP</span></span>
<span><span class="co">#&gt; [1] "ABCA1"   "ACTN1"   "JAG1"    "ALOX5AP" "APBB2"   "ARNTL"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $FOROUTAN_TGFB_EMT_DN</span></span>
<span><span class="co">#&gt; [1] "ADORA2B" "ALDH1A3" "ALDH3A2" "ANK3"    "BIRC3"   "AQP3"   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $FOROUTAN_PRODRANK_TGFB_EMT_UP</span></span>
<span><span class="co">#&gt; [1] "ABCA1"   "ACTN1"   "JAG1"    "ALOX5AP" "APBB2"   "ARNTL"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $FOROUTAN_PRODRANK_TGFB_EMT_DN</span></span>
<span><span class="co">#&gt; [1] "ADORA2B" "ALDH3A2" "ANK3"    "BIRC3"   "AREG"    "DST"    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $FOROUTAN_INTEGRATED_TGFB_EMT_UP</span></span>
<span><span class="co">#&gt; [1] "ALOX5AP" "BMPR2"   "CALD1"   "CDH2"    "CDH11"   "COL1A1"</span></span></code></pre></div>
</div>
<div class="section level2" number="5">
<h2 id="session-information">
<span class="header-section-number">5</span> Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] GSVA_1.53.17         GSEABase_1.67.0      graph_1.83.0        </span></span>
<span><span class="co">#&gt;  [4] annotate_1.83.0      XML_3.99-0.17        AnnotationDbi_1.67.0</span></span>
<span><span class="co">#&gt;  [7] IRanges_2.39.2       S4Vectors_0.43.2     Biobase_2.65.1      </span></span>
<span><span class="co">#&gt; [10] BiocGenerics_0.51.0 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] SummarizedExperiment_1.35.1 KEGGREST_1.45.1            </span></span>
<span><span class="co">#&gt;  [3] rjson_0.2.22                xfun_0.47                  </span></span>
<span><span class="co">#&gt;  [5] bslib_0.8.0                 htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt;  [7] rhdf5_2.49.0                lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [9] rhdf5filters_1.17.0         vctrs_0.6.5                </span></span>
<span><span class="co">#&gt; [11] tools_4.4.1                 parallel_4.4.1             </span></span>
<span><span class="co">#&gt; [13] RSQLite_2.3.7               blob_1.2.4                 </span></span>
<span><span class="co">#&gt; [15] Matrix_1.7-0                sparseMatrixStats_1.17.2   </span></span>
<span><span class="co">#&gt; [17] lifecycle_1.0.4             GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">#&gt; [19] compiler_4.4.1              Biostrings_2.73.1          </span></span>
<span><span class="co">#&gt; [21] codetools_0.2-20            GenomeInfoDb_1.41.1        </span></span>
<span><span class="co">#&gt; [23] htmltools_0.5.8.1           sass_0.4.9                 </span></span>
<span><span class="co">#&gt; [25] yaml_2.3.10                 crayon_1.5.3               </span></span>
<span><span class="co">#&gt; [27] jquerylib_0.1.4             SingleCellExperiment_1.27.2</span></span>
<span><span class="co">#&gt; [29] BiocParallel_1.39.0         DelayedArray_0.31.11       </span></span>
<span><span class="co">#&gt; [31] cachem_1.1.0                magick_2.8.4               </span></span>
<span><span class="co">#&gt; [33] abind_1.4-5                 rsvd_1.0.5                 </span></span>
<span><span class="co">#&gt; [35] digest_0.6.37               BiocSingular_1.21.2        </span></span>
<span><span class="co">#&gt; [37] bookdown_0.40               fastmap_1.2.0              </span></span>
<span><span class="co">#&gt; [39] grid_4.4.1                  cli_3.6.3                  </span></span>
<span><span class="co">#&gt; [41] SparseArray_1.5.31          magrittr_2.0.3             </span></span>
<span><span class="co">#&gt; [43] S4Arrays_1.5.7              UCSC.utils_1.1.0           </span></span>
<span><span class="co">#&gt; [45] bit64_4.0.5                 rmarkdown_2.28             </span></span>
<span><span class="co">#&gt; [47] XVector_0.45.0              httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [49] matrixStats_1.3.0           bit_4.0.5                  </span></span>
<span><span class="co">#&gt; [51] png_0.1-8                   SpatialExperiment_1.15.1   </span></span>
<span><span class="co">#&gt; [53] ScaledMatrix_1.13.0         beachmat_2.21.5            </span></span>
<span><span class="co">#&gt; [55] memoise_2.0.1               HDF5Array_1.33.6           </span></span>
<span><span class="co">#&gt; [57] evaluate_0.24.0             knitr_1.48                 </span></span>
<span><span class="co">#&gt; [59] GenomicRanges_1.57.1        irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt; [61] rlang_1.1.4                 Rcpp_1.0.13                </span></span>
<span><span class="co">#&gt; [63] xtable_1.8-4                DBI_1.2.3                  </span></span>
<span><span class="co">#&gt; [65] jsonlite_1.8.8              Rhdf5lib_1.27.0            </span></span>
<span><span class="co">#&gt; [67] R6_2.5.1                    MatrixGenerics_1.17.0      </span></span>
<span><span class="co">#&gt; [69] zlibbioc_1.51.1</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-subramanian2005gene" class="csl-entry">
Subramanian, Aravind, Pablo Tamayo, Vamsi K Mootha, Sayan Mukherjee, Benjamin L Ebert, Michael A Gillette, Amanda Paulovich, et al. 2005. <span>“Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.”</span> <em>Proceedings of the National Academy of Sciences</em> 102 (43): 15545–50. <a href="https://doi.org/10.1073/pnas.0506580102" class="external-link">https://doi.org/10.1073/pnas.0506580102</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robert Castelo, Axel Klenk.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
