<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of single-cell RNA-seq data â€¢ GSVAEuroBioC2024</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of single-cell RNA-seq data">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSVAEuroBioC2024</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/importGMTfiles.html">Import gene sets from GMT files</a>
    </li>
    <li>
      <a href="../articles/bulkRNAseq.html">Analysis of bulk RNA-seq data</a>
    </li>
    <li>
      <a href="../articles/scRNAseq.html">Analysis of single-cell RNA-seq data</a>
    </li>
    <li>
      <a href="../articles/spatial.html">Analysis of spatial omics data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/functionalgenomics/GSVAEuroBioC2024" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of single-cell RNA-seq data</h1>
                        <h4 data-toc-skip class="author">Axel Klenk</h4>
            <address class="author_afil">
      Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona<br><a class="author_email" href="mailto:#"></a><a href="mailto:axelvolker.klenk@upf.edu" class="email">axelvolker.klenk@upf.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Robert Castelo</h4>
            <address class="author_afil">
      Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona<br><a class="author_email" href="mailto:#"></a><a href="mailto:robert.castelo@upf.edu" class="email">robert.castelo@upf.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">2024-09-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/functionalgenomics/GSVAEuroBioC2024/blob/devel/vignettes/scRNAseq.Rmd" class="external-link"><code>vignettes/scRNAseq.Rmd</code></a></small>
      <div class="hidden name"><code>scRNAseq.Rmd</code></div>

    </div>

    
    
<p>Vignette built using <code>GSVA</code>, version 1.53.16.</p>
<div class="section level2" number="1">
<h2 id="importing-scrna-seq-data-into-a-singlecellexperiment-object">
<span class="header-section-number">1</span> Importing scRNA-seq data into a <code>SingleCellExperiment</code> object<a class="anchor" aria-label="anchor" href="#importing-scrna-seq-data-into-a-singlecellexperiment-object"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TENxPBMCData</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TENxPBMCData/man/TENxPBMCData.html" class="external-link">TENxPBMCData</a></span><span class="op">(</span>dataset<span class="op">=</span><span class="st">"pbmc4k"</span><span class="op">)</span></span>
<span><span class="va">sce</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 33694 4340 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(33694): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span></span>
<span><span class="co">#&gt;   ENSG00000268674</span></span>
<span><span class="co">#&gt; rowData names(3): ENSEMBL_ID Symbol_TENx Symbol</span></span>
<span><span class="co">#&gt; colnames: NULL</span></span>
<span><span class="co">#&gt; colData names(11): Sample Barcode ... Individual Date_published</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
</div>
<div class="section level2" number="2">
<h2 id="importing-gene-sets">
<span class="header-section-number">2</span> Importing gene sets<a class="anchor" aria-label="anchor" href="#importing-gene-sets"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rcastelo/GSVA" class="external-link">GSVA</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">fname</span> <span class="op">&lt;-</span> <span class="st">"PBMC_gene_sets_TrianaEtAl2021.gmt"</span></span>
<span><span class="va">gsets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">fname</span>, package<span class="op">=</span><span class="st">"GSVAEuroBioC2024"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 id="how-to-run-gsva">
<span class="header-section-number">3</span> How to run GSVA<a class="anchor" aria-label="anchor" href="#how-to-run-gsva"></a>
</h2>
</div>
<div class="section level2" number="4">
<h2 id="session-information">
<span class="header-section-number">4</span> Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] GSVA_1.53.16                TENxPBMCData_1.23.0        </span></span>
<span><span class="co">#&gt;  [3] HDF5Array_1.33.6            rhdf5_2.49.0               </span></span>
<span><span class="co">#&gt;  [5] DelayedArray_0.31.11        SparseArray_1.5.31         </span></span>
<span><span class="co">#&gt;  [7] S4Arrays_1.5.7              abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [9] Matrix_1.7-0                SingleCellExperiment_1.27.2</span></span>
<span><span class="co">#&gt; [11] SummarizedExperiment_1.35.1 Biobase_2.65.1             </span></span>
<span><span class="co">#&gt; [13] GenomicRanges_1.57.1        GenomeInfoDb_1.41.1        </span></span>
<span><span class="co">#&gt; [15] IRanges_2.39.2              S4Vectors_0.43.2           </span></span>
<span><span class="co">#&gt; [17] BiocGenerics_0.51.0         MatrixGenerics_1.17.0      </span></span>
<span><span class="co">#&gt; [19] matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.1         dplyr_1.1.4              blob_1.2.4              </span></span>
<span><span class="co">#&gt;  [4] filelock_1.0.3           Biostrings_2.73.1        fastmap_1.2.0           </span></span>
<span><span class="co">#&gt;  [7] BiocFileCache_2.13.0     XML_3.99-0.17            rsvd_1.0.5              </span></span>
<span><span class="co">#&gt; [10] digest_0.6.37            mime_0.12                lifecycle_1.0.4         </span></span>
<span><span class="co">#&gt; [13] KEGGREST_1.45.1          RSQLite_2.3.7            magrittr_2.0.3          </span></span>
<span><span class="co">#&gt; [16] compiler_4.4.1           rlang_1.1.4              sass_0.4.9              </span></span>
<span><span class="co">#&gt; [19] tools_4.4.1              utf8_1.2.4               yaml_2.3.10             </span></span>
<span><span class="co">#&gt; [22] knitr_1.48               htmlwidgets_1.6.4        bit_4.0.5               </span></span>
<span><span class="co">#&gt; [25] curl_5.2.2               BiocParallel_1.39.0      withr_3.0.1             </span></span>
<span><span class="co">#&gt; [28] purrr_1.0.2              grid_4.4.1               fansi_1.0.6             </span></span>
<span><span class="co">#&gt; [31] ExperimentHub_2.13.1     beachmat_2.21.5          xtable_1.8-4            </span></span>
<span><span class="co">#&gt; [34] Rhdf5lib_1.27.0          cli_3.6.3                rmarkdown_2.28          </span></span>
<span><span class="co">#&gt; [37] crayon_1.5.3             generics_0.1.3           rjson_0.2.22            </span></span>
<span><span class="co">#&gt; [40] httr_1.4.7               DBI_1.2.3                cachem_1.1.0            </span></span>
<span><span class="co">#&gt; [43] zlibbioc_1.51.1          parallel_4.4.1           AnnotationDbi_1.67.0    </span></span>
<span><span class="co">#&gt; [46] BiocManager_1.30.25      XVector_0.45.0           vctrs_0.6.5             </span></span>
<span><span class="co">#&gt; [49] jsonlite_1.8.8           BiocSingular_1.21.2      bookdown_0.40           </span></span>
<span><span class="co">#&gt; [52] bit64_4.0.5              GSEABase_1.67.0          irlba_2.3.5.1           </span></span>
<span><span class="co">#&gt; [55] magick_2.8.4             jquerylib_0.1.4          annotate_1.83.0         </span></span>
<span><span class="co">#&gt; [58] glue_1.7.0               codetools_0.2-20         BiocVersion_3.20.0      </span></span>
<span><span class="co">#&gt; [61] UCSC.utils_1.1.0         ScaledMatrix_1.13.0      tibble_3.2.1            </span></span>
<span><span class="co">#&gt; [64] pillar_1.9.0             rappdirs_0.3.3           htmltools_0.5.8.1       </span></span>
<span><span class="co">#&gt; [67] rhdf5filters_1.17.0      graph_1.83.0             GenomeInfoDbData_1.2.12 </span></span>
<span><span class="co">#&gt; [70] R6_2.5.1                 dbplyr_2.5.0             sparseMatrixStats_1.17.2</span></span>
<span><span class="co">#&gt; [73] evaluate_0.24.0          lattice_0.22-6           AnnotationHub_3.13.3    </span></span>
<span><span class="co">#&gt; [76] png_0.1-8                SpatialExperiment_1.15.1 memoise_2.0.1           </span></span>
<span><span class="co">#&gt; [79] bslib_0.8.0              Rcpp_1.0.13              xfun_0.47               </span></span>
<span><span class="co">#&gt; [82] pkgconfig_2.0.3</span></span></code></pre></div>
</div>
<div class="section level2" number="5">
<h2 id="references">
<span class="header-section-number">5</span> References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robert Castelo, Axel Klenk.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
