<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of single-cell RNA-seq data • GSVAEuroBioC2024</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of single-cell RNA-seq data">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSVAEuroBioC2024</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/importGMTfiles.html">Import gene sets from GMT files</a>
    </li>
    <li>
      <a href="../articles/bulkRNAseq.html">Analysis of bulk RNA-seq data</a>
    </li>
    <li>
      <a href="../articles/scRNAseq.html">Analysis of single-cell RNA-seq data</a>
    </li>
    <li>
      <a href="../articles/spatial.html">Analysis of spatial omics data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/functionalgenomics/GSVAEuroBioC2024" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of single-cell RNA-seq data</h1>
                        <h4 data-toc-skip class="author">Axel Klenk</h4>
            <address class="author_afil">
      Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona<br><a class="author_email" href="mailto:#"></a><a href="mailto:axelvolker.klenk@upf.edu" class="email">axelvolker.klenk@upf.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Robert Castelo</h4>
            <address class="author_afil">
      Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra, Barcelona<br><a class="author_email" href="mailto:#"></a><a href="mailto:robert.castelo@upf.edu" class="email">robert.castelo@upf.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">2024-09-04</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/functionalgenomics/GSVAEuroBioC2024/blob/devel/vignettes/scRNAseq.Rmd" class="external-link"><code>vignettes/scRNAseq.Rmd</code></a></small>
      <div class="hidden name"><code>scRNAseq.Rmd</code></div>

    </div>

    
    
<p>Vignette built using <code>GSVA</code>, version 1.53.19.</p>
<div class="section level2" number="1">
<h2 id="importing-scrna-seq-data-into-a-singlecellexperiment-object">
<span class="header-section-number">1</span> Importing scRNA-seq data into a <code>SingleCellExperiment</code> object<a class="anchor" aria-label="anchor" href="#importing-scrna-seq-data-into-a-singlecellexperiment-object"></a>
</h2>
<p>In this vignette we will use the
<a href="https://bioconductor.org/packages/TENxPBMCData" class="external-link">TENxPBMCData</a> to download a
<a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets" class="external-link">10X Genomics</a>
single-cell RNA-seq data set generated from peripheral blood mononuclear cells
(PBMC), into a
<a href="https://bioconductor.org/packages/SingleCellExperiment" class="external-link">SingleCellExperiment</a>
object.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TENxPBMCData</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TENxPBMCData/man/TENxPBMCData.html" class="external-link">TENxPBMCData</a></span><span class="op">(</span>dataset<span class="op">=</span><span class="st">"pbmc4k"</span><span class="op">)</span></span>
<span><span class="va">sce</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 33694 4340 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(33694): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span></span>
<span><span class="co">#&gt;   ENSG00000268674</span></span>
<span><span class="co">#&gt; rowData names(3): ENSEMBL_ID Symbol_TENx Symbol</span></span>
<span><span class="co">#&gt; colnames: NULL</span></span>
<span><span class="co">#&gt; colData names(11): Sample Barcode ... Individual Date_published</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
<p>This single-cell dataset consists of raw counts for 33694 genes by
4340 cells.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span></span>
<span><span class="co">#&gt; &lt;5 x 8&gt; sparse DelayedMatrix object of type "integer":</span></span>
<span><span class="co">#&gt;                 [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8]</span></span>
<span><span class="co">#&gt; ENSG00000243485    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; ENSG00000237613    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; ENSG00000186092    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; ENSG00000238009    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; ENSG00000239945    0    0    0    0    0    0    0    0</span></span></code></pre></div>
</div>
<div class="section level2" number="2">
<h2 id="pre-processing-of-scrna-seq-data">
<span class="header-section-number">2</span> Pre-processing of scRNA-seq data<a class="anchor" aria-label="anchor" href="#pre-processing-of-scrna-seq-data"></a>
</h2>
<p>Here we do the pre-processing of the scRNA-seq dataset using the
Bioconductor package <a href="https://bioconductor.org/packages/scuttle" class="external-link">scuttle</a>. We
start by identifying mitochondrial genes.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scuttle</span><span class="op">)</span></span>
<span></span>
<span><span class="va">is_mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"^MT-"</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">Symbol_TENx</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">is_mito</span><span class="op">)</span> </span>
<span><span class="co">#&gt; is_mito</span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt; 33681    13</span></span></code></pre></div>
<p>Now calculate quality control (QC) metrics and filter out low-quality cells.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/quickPerCellQC.html" class="external-link">quickPerCellQC</a></span><span class="op">(</span><span class="va">sce</span>, subsets<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Mito<span class="op">=</span><span class="va">is_mito</span><span class="op">)</span>,</span>
<span>                      sub.fields<span class="op">=</span><span class="st">"subsets_Mito_percent"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 33694  4147</span></span></code></pre></div>
<p>Figure <a href="#fig:cntxgene">2.1</a> below shows the empirical cumulative distribution of
counts per gene in logarithmic scale.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cntxgene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">+</span><span class="fl">1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/ecdf.html" class="external-link">plot.ecdf</a></span><span class="op">(</span><span class="va">cntxgene</span>, xaxt<span class="op">=</span><span class="st">"n"</span>, panel.first<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/graphics/grid.html" class="external-link">grid</a></span><span class="op">(</span><span class="op">)</span>, xlab<span class="op">=</span><span class="st">"UMI counts per gene"</span>,</span>
<span>                    log<span class="op">=</span><span class="st">"x"</span>, main<span class="op">=</span><span class="st">""</span>, xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1e5</span><span class="op">)</span>, las<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span><span class="fl">1</span>, at<span class="op">=</span><span class="fl">10</span><span class="op">^</span><span class="op">(</span><span class="fl">0</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span>, labels<span class="op">=</span><span class="fl">10</span><span class="op">^</span><span class="op">(</span><span class="fl">0</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v<span class="op">=</span><span class="fl">100</span>, lwd<span class="op">=</span><span class="fl">2</span>, col<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:cntxgene"></span>
<img src="scRNAseq_files/figure-html/cntxgene-1.png" alt="Filtering lowly-expressed genes. Empirical cumulative distribution of UMI counts per gene. The red vertical bar indicates a cutoff value of 100 UMI counts per gene across all cells, below which genes will be filtered out."><p class="caption">
Figure 2.1: Filtering lowly-expressed genes. Empirical cumulative distribution of UMI counts per gene. The red vertical bar indicates a cutoff value of 100 UMI counts per gene across all cells, below which genes will be filtered out.
</p>
</div>
<p>We filter out lowly-expressed genes, by selecting those with at least 100 UMI
counts across all cells.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span><span class="va">cntxgene</span> <span class="op">&gt;=</span> <span class="fl">100</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 8823 4147</span></span></code></pre></div>
<p>Finally, we calculate library size factors and normalized log units of
expression.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/librarySizeFactors.html" class="external-link">computeLibraryFactors</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "counts"    "logcounts"</span></span></code></pre></div>
<p>The assay <code>logcounts</code> contains the filtered and normalized log units of
expression that we should use as input to GSVA.</p>
</div>
<div class="section level2" number="3">
<h2 id="importing-gene-sets">
<span class="header-section-number">3</span> Importing gene sets<a class="anchor" aria-label="anchor" href="#importing-gene-sets"></a>
</h2>
<p>Here we read a GMT file storing a collection of 22 gene sets used by
<span class="citation">Diaz-Mejia et al. (<a href="#ref-diaz2019evaluation">2019</a>)</span> to benchmark methods for cell type assignment in PBMC
scRNA-seq data.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rcastelo/GSVA" class="external-link">GSVA</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"pbmc_cell_type_gene_set_signatures.gmt.gz"</span>,</span>
<span>                     package<span class="op">=</span><span class="st">"GSVAEuroBioC2024"</span><span class="op">)</span></span>
<span><span class="va">genesets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/readGMT.html" class="external-link">readGMT</a></span><span class="op">(</span><span class="va">fname</span>, geneIdType<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">genesets</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: B_CELLS_MEMORY, B_CELLS_NAIVE, ..., T_CELLS_REGULATORY_TREGS (22 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: AIM2, BANK1, ..., SKAP1 (248 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: NullCollection (1 total)</span></span></code></pre></div>
</div>
<div class="section level2" number="4">
<h2 id="running-gsva-on-a-singlecellexperiment-object">
<span class="header-section-number">4</span> Running GSVA on a <code>SingleCellExperiment</code> object<a class="anchor" aria-label="anchor" href="#running-gsva-on-a-singlecellexperiment-object"></a>
</h2>
<p>GSVA can take as input a <code>SingleCellExperiment</code> object, but before we run
GSVA we should take a couple of steps and considerations. In this vignette,
genes in that object are defined by
<a href="https://www.ensembl.org/info/genome/stable_ids/index.html" class="external-link">Ensembl stable gene identifiers</a>,
i.e., they start with <code>ENSG</code>, while gene identifiers in the collection of genes
we imported in the previous section correspond to
<a href="https://www.genenames.org" class="external-link">HUGO gene symbols</a>. To facilitate that GSVA
internally maps the gene symbols in gene sets to the corresponding Ensembl gene
identifiers of the scRNA-seq data, we should add the following metadata to the
<code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/gsvaAnnotation.html" class="external-link">gsvaAnnotation</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">ENSEMBLIdentifier</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span></code></pre></div>
<p>The assay data is stored as a sparse <code>DelayedMatrix</code> object using an <code>HDF5</code>
backend. However, GSVA currently<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> only provides support for sparse
matrices stored in <code>dgCMatrix</code> objects.</p>
<p>This means that we need to coerce the assay we intend to use as input for
GSVA to a <code>dgCMatrix</code>, as follows.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span>, <span class="st">"dgCMatrix"</span><span class="op">)</span></span></code></pre></div>
<p>Independently of this previous issue, one should be aware of the name of the
assay that contains the normalized expression values we want to provide as
input to GSVA. In this case this assay is called <code>logcounts</code>. Now we can
perform the first step to run GSVA, which consists of building a parameter
object.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsvapar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/gsvaParam-class.html" class="external-link">gsvaParam</a></span><span class="op">(</span><span class="va">sce</span>, <span class="va">genesets</span>, assay<span class="op">=</span><span class="st">"logcounts"</span><span class="op">)</span></span>
<span><span class="va">gsvapar</span></span>
<span><span class="co">#&gt; A GSVA::gsvaParam object</span></span>
<span><span class="co">#&gt; expression data:</span></span>
<span><span class="co">#&gt;   class: SingleCellExperiment </span></span>
<span><span class="co">#&gt;   dim: 8823 4147 </span></span>
<span><span class="co">#&gt;   metadata(1): annotation</span></span>
<span><span class="co">#&gt;   assays(2): counts logcounts</span></span>
<span><span class="co">#&gt;   rownames(8823): ENSG00000279457 ENSG00000228463 ... ENSG00000198727</span></span>
<span><span class="co">#&gt;     ENSG00000273748</span></span>
<span><span class="co">#&gt;   rowData names(3): ENSEMBL_ID Symbol_TENx Symbol</span></span>
<span><span class="co">#&gt;   colnames: NULL</span></span>
<span><span class="co">#&gt;   colData names(22): Sample Barcode ... discard sizeFactor</span></span>
<span><span class="co">#&gt;   reducedDimNames(0):</span></span>
<span><span class="co">#&gt;   mainExpName: NULL</span></span>
<span><span class="co">#&gt;   altExpNames(0):</span></span>
<span><span class="co">#&gt; using assay: logcounts</span></span>
<span><span class="co">#&gt; using annotation: none</span></span>
<span><span class="co">#&gt; gene sets:</span></span>
<span><span class="co">#&gt;   GeneSetCollection</span></span>
<span><span class="co">#&gt;     names: B_CELLS_MEMORY, B_CELLS_NAIVE, ..., T_CELLS_REGULATORY_TREGS (22 total)</span></span>
<span><span class="co">#&gt;     unique identifiers: AIM2, BANK1, ..., SKAP1 (248 total)</span></span>
<span><span class="co">#&gt;     types in collection:</span></span>
<span><span class="co">#&gt;       geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;       collectionType: NullCollection (1 total)</span></span>
<span><span class="co">#&gt; gene set size: [1, Inf]</span></span>
<span><span class="co">#&gt; kcdf: auto</span></span>
<span><span class="co">#&gt; kcdfNoneMinSampleSize: 50</span></span>
<span><span class="co">#&gt; tau: 1</span></span>
<span><span class="co">#&gt; maxDiff: TRUE</span></span>
<span><span class="co">#&gt; absRanking: FALSE</span></span>
<span><span class="co">#&gt; sparse: TRUE</span></span></code></pre></div>
<p>In the second and final step, we call the function <code><a href="https://rdrr.io/pkg/GSVA/man/gsva.html" class="external-link">gsva()</a></code> with the previous
parameter object as input.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">es</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSVA/man/gsva.html" class="external-link">gsva</a></span><span class="op">(</span><span class="va">gsvapar</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> GSVA version 1.53.19</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Mapping identifiers</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Calculating GSVA scores for 22 gene sets</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> kcdf='auto' (default)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> GSVA sparse algorithm</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Direct row-wise ECDFs estimation</span></span>
<span><span class="co">#&gt; Estimating ECDFs <span style="color: #00BB00;">■■■■■■■■■■■■■■■                 </span>  48% |  ETA:  1s</span></span>
<span><span class="co">#&gt; Calculating GSVA scores <span style="color: #00BB00;">■■■■                            </span>   9% | ETA: 13s</span></span>
<span><span class="co">#&gt; Calculating GSVA scores <span style="color: #00BB00;">■■■■■■■■■■■                     </span>  32% | ETA:  9s</span></span>
<span><span class="co">#&gt; Calculating GSVA scores <span style="color: #00BB00;">■■■■■■■■■■■■■■■■■■              </span>  56% | ETA:  6s</span></span>
<span><span class="co">#&gt; Calculating GSVA scores <span style="color: #00BB00;">■■■■■■■■■■■■■■■■■■■■■■■■■       </span>  79% | ETA:  3s</span></span>
<span><span class="co">#&gt; Calculating GSVA scores <span style="color: #00BB00;">■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■ </span> 100% | ETA:  0s</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Calculations finished</span></span></code></pre></div>
</div>
<div class="section level2" number="5">
<h2 id="using-gsva-scores-to-assign-cell-types">
<span class="header-section-number">5</span> Using GSVA scores to assign cell types<a class="anchor" aria-label="anchor" href="#using-gsva-scores-to-assign-cell-types"></a>
</h2>
<p>Following <span class="citation">(<a href="#ref-amezquita2020orchestrating">Amezquita et al. 2020</a>)</span>, and some of the steps described in Chapter
5 Clustering of the
<a href="https://bioconductor.org/books/3.16/OSCA/clustering.html" class="external-link">OSCA book</a>, here we
use GSVA scores to create first a nearst-neighbor graph of the cells using the
function <code><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html" class="external-link">buildSNNGraph()</a></code> from the Bioconductor package
<a href="https://bioconductor.org/packages/scran" class="external-link">scran</a>, and then use this graph to
cluster the cells using the community detection algorithm implemented in the
function <code><a href="https://r.igraph.org/reference/cluster_walktrap.html" class="external-link">cluster_walktrap()</a></code> from the CRAN package
<a href="https://bioconductor.org/packages/igraph" class="external-link">igraph</a>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MarioniLab/scran/" class="external-link">scran</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html" class="external-link">buildSNNGraph</a></span><span class="op">(</span><span class="va">es</span>, k<span class="op">=</span><span class="fl">20</span>, assay.type<span class="op">=</span><span class="st">"es"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html" class="external-link">colLabels</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://r.igraph.org/reference/cluster_walktrap.html" class="external-link">cluster_walktrap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">membership</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html" class="external-link">colLabels</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    1    2    3    4 </span></span>
<span><span class="co">#&gt;  780 1040  618 1709</span></span></code></pre></div>
<p>We set <code>k=20</code> in the call to <code><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html" class="external-link">buildSNNGraph()</a></code> so that we get a number of
clusters close to the number of cell types found in PBMCs (T cells, B cells,
NK cells, monocytes and dendritic cells).</p>
<p>Similarly to <span class="citation">Diaz-Mejia et al. (<a href="#ref-diaz2019evaluation">2019</a>)</span>, we apply a very simple cell type assignment
algorithm, which consists of selecting at each cell the gene set with highest
GSVA score, tallying the selected gene sets per cluster, and assigning to the
cluster the most frequent gene set, storing that assignment in the column data
of the <code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">whmax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span>, <span class="fl">2</span>, <span class="va">which.max</span><span class="op">)</span></span>
<span><span class="va">gsetsxlabels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">[</span><span class="va">whmax</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html" class="external-link">colLabels</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gsetsxlabels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gsetsxlabels</span>, <span class="va">table</span><span class="op">)</span>, <span class="va">which.max</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">$</span><span class="va">gsetsxlabel</span> <span class="op">&lt;-</span> <span class="va">gsetsxlabels</span><span class="op">[</span><span class="va">es</span><span class="op">$</span><span class="va">label</span><span class="op">]</span></span></code></pre></div>
<p>Finally, using the Bioconductor package
<a href="https://bioconductor.org/packages/scater" class="external-link">scater</a>, we calculate the first
two PCA components on the GSVA scores and use them for visualizing in Figure
<a href="#fig:gsvaclusters">5.1</a> the similarity between cells in terms of GSVA scores
project in two dimensions, the clusters based on the community detection
algorithm and the cell type assignments based on gene sets.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">es</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">es</span>, assay.type<span class="op">=</span><span class="st">"es"</span>, ncomponents<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span><span class="va">es</span>, dimred<span class="op">=</span><span class="st">"PCA"</span>, colour_by<span class="op">=</span><span class="st">"label"</span>, text_by<span class="op">=</span><span class="st">"gsetsxlabel"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:gsvaclusters"></span>
<img src="scRNAseq_files/figure-html/gsvaclusters-1.png" alt="Cell type assignments of PBMC scRNA-seq data, based on GSVA scores."><p class="caption">
Figure 5.1: Cell type assignments of PBMC scRNA-seq data, based on GSVA scores.
</p>
</div>
</div>
<div class="section level2" number="6">
<h2 id="session-information">
<span class="header-section-number">6</span> Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] scater_1.33.4               ggplot2_3.5.1              </span></span>
<span><span class="co">#&gt;  [3] igraph_2.0.3                scran_1.33.1               </span></span>
<span><span class="co">#&gt;  [5] org.Hs.eg.db_3.19.1         GSVA_1.53.19               </span></span>
<span><span class="co">#&gt;  [7] GSEABase_1.67.0             graph_1.83.0               </span></span>
<span><span class="co">#&gt;  [9] annotate_1.83.0             XML_3.99-0.17              </span></span>
<span><span class="co">#&gt; [11] AnnotationDbi_1.67.0        scuttle_1.15.4             </span></span>
<span><span class="co">#&gt; [13] TENxPBMCData_1.23.0         HDF5Array_1.33.6           </span></span>
<span><span class="co">#&gt; [15] rhdf5_2.49.0                DelayedArray_0.31.11       </span></span>
<span><span class="co">#&gt; [17] SparseArray_1.5.31          S4Arrays_1.5.7             </span></span>
<span><span class="co">#&gt; [19] abind_1.4-5                 Matrix_1.7-0               </span></span>
<span><span class="co">#&gt; [21] SingleCellExperiment_1.27.2 SummarizedExperiment_1.35.1</span></span>
<span><span class="co">#&gt; [23] Biobase_2.65.1              GenomicRanges_1.57.1       </span></span>
<span><span class="co">#&gt; [25] GenomeInfoDb_1.41.1         IRanges_2.39.2             </span></span>
<span><span class="co">#&gt; [27] S4Vectors_0.43.2            BiocGenerics_0.51.0        </span></span>
<span><span class="co">#&gt; [29] MatrixGenerics_1.17.0       matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] jsonlite_1.8.8           magrittr_2.0.3           ggbeeswarm_0.7.2        </span></span>
<span><span class="co">#&gt;   [4] magick_2.8.4             farver_2.1.2             rmarkdown_2.28          </span></span>
<span><span class="co">#&gt;   [7] zlibbioc_1.51.1          vctrs_0.6.5              memoise_2.0.1           </span></span>
<span><span class="co">#&gt;  [10] htmltools_0.5.8.1        AnnotationHub_3.13.3     curl_5.2.2              </span></span>
<span><span class="co">#&gt;  [13] BiocNeighbors_1.23.0     Rhdf5lib_1.27.0          sass_0.4.9              </span></span>
<span><span class="co">#&gt;  [16] bslib_0.8.0              htmlwidgets_1.6.4        cachem_1.1.0            </span></span>
<span><span class="co">#&gt;  [19] mime_0.12                lifecycle_1.0.4          pkgconfig_2.0.3         </span></span>
<span><span class="co">#&gt;  [22] rsvd_1.0.5               R6_2.5.1                 fastmap_1.2.0           </span></span>
<span><span class="co">#&gt;  [25] GenomeInfoDbData_1.2.12  digest_0.6.37            colorspace_2.1-1        </span></span>
<span><span class="co">#&gt;  [28] dqrng_0.4.1              irlba_2.3.5.1            ExperimentHub_2.13.1    </span></span>
<span><span class="co">#&gt;  [31] RSQLite_2.3.7            beachmat_2.21.5          labeling_0.4.3          </span></span>
<span><span class="co">#&gt;  [34] filelock_1.0.3           fansi_1.0.6              httr_1.4.7              </span></span>
<span><span class="co">#&gt;  [37] compiler_4.4.1           bit64_4.0.5              withr_3.0.1             </span></span>
<span><span class="co">#&gt;  [40] BiocParallel_1.39.0      viridis_0.6.5            DBI_1.2.3               </span></span>
<span><span class="co">#&gt;  [43] highr_0.11               rappdirs_0.3.3           rjson_0.2.22            </span></span>
<span><span class="co">#&gt;  [46] bluster_1.15.0           tools_4.4.1              vipor_0.4.7             </span></span>
<span><span class="co">#&gt;  [49] beeswarm_0.4.0           glue_1.7.0               rhdf5filters_1.17.0     </span></span>
<span><span class="co">#&gt;  [52] grid_4.4.1               cluster_2.1.6            generics_0.1.3          </span></span>
<span><span class="co">#&gt;  [55] gtable_0.3.5             BiocSingular_1.21.2      ScaledMatrix_1.13.0     </span></span>
<span><span class="co">#&gt;  [58] metapod_1.13.0           utf8_1.2.4               XVector_0.45.0          </span></span>
<span><span class="co">#&gt;  [61] ggrepel_0.9.5            BiocVersion_3.20.0       pillar_1.9.0            </span></span>
<span><span class="co">#&gt;  [64] limma_3.61.9             dplyr_1.1.4              BiocFileCache_2.13.0    </span></span>
<span><span class="co">#&gt;  [67] lattice_0.22-6           bit_4.0.5                tidyselect_1.2.1        </span></span>
<span><span class="co">#&gt;  [70] locfit_1.5-9.10          Biostrings_2.73.1        knitr_1.48              </span></span>
<span><span class="co">#&gt;  [73] gridExtra_2.3            bookdown_0.40            edgeR_4.3.14            </span></span>
<span><span class="co">#&gt;  [76] xfun_0.47                statmod_1.5.0            UCSC.utils_1.1.0        </span></span>
<span><span class="co">#&gt;  [79] yaml_2.3.10              evaluate_0.24.0          codetools_0.2-20        </span></span>
<span><span class="co">#&gt;  [82] tibble_3.2.1             BiocManager_1.30.25      cli_3.6.3               </span></span>
<span><span class="co">#&gt;  [85] xtable_1.8-4             munsell_0.5.1            jquerylib_0.1.4         </span></span>
<span><span class="co">#&gt;  [88] Rcpp_1.0.13              dbplyr_2.5.0             png_0.1-8               </span></span>
<span><span class="co">#&gt;  [91] parallel_4.4.1           blob_1.2.4               sparseMatrixStats_1.17.2</span></span>
<span><span class="co">#&gt;  [94] SpatialExperiment_1.15.1 viridisLite_0.4.2        scales_1.3.0            </span></span>
<span><span class="co">#&gt;  [97] purrr_1.0.2              crayon_1.5.3             rlang_1.1.4             </span></span>
<span><span class="co">#&gt; [100] KEGGREST_1.45.1</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-amezquita2020orchestrating" class="csl-entry">
Amezquita, Robert A, Aaron TL Lun, Etienne Becht, Vince J Carey, Lindsay N Carpp, Ludwig Geistlinger, Federico Marini, et al. 2020. <span>“Orchestrating Single-Cell Analysis with Bioconductor.”</span> <em>Nature Methods</em> 17 (2): 137–45. <a href="https://doi.org/10.1038/s41592-019-0654-x" class="external-link">https://doi.org/10.1038/s41592-019-0654-x</a>.
</div>
<div id="ref-diaz2019evaluation" class="csl-entry">
Diaz-Mejia, J Javier, Elaine C Meng, Alexander R Pico, Sonya A MacParland, Troy Ketela, Trevor J Pugh, Gary D Bader, and John H Morris. 2019. <span>“Evaluation of Methods to Assign Cell Type Labels to Cell Clusters from Single-Cell RNA-Sequencing Data.”</span> <em>F1000Research</em> 8. <a href="https://doi.org/10.12688/f1000research.18490.3" class="external-link">https://doi.org/10.12688/f1000research.18490.3</a>.
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p>We expect to provide support in the near future for sparse
on-disk matrices, such as sparse <code>DelayedMatrix</code> objects.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robert Castelo, Axel Klenk.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
